---
title: "Script for data paper: Root-associated microbiota of decline-affected and asymptomatic *Pinus sylvestris* trees"
bibliography: references.bib
---

This repository contains scripts and analysis used for the datapaper "Root-associated microbiota of decline-affected and asymptomatic *Pinus sylvestris* trees" 

# Introduction  

The SUMHAL Project aims at deciphering the changes in the microbial communities associated to the roots of different *Pinus* species as a consequence of forest decline. Three different mountainous areas covered by *Pinus* spp. stands in the Southeast Spain were selected for this purpose.
Since different pine species in different conditions were selected, different sub-projects were created, which resulted in different scientific publications. Below you will find specific information about the main project and sub-projects

# Experimental design

In the following table, all the information regarding the project is summarized 

| Species                             | Season | Plant compartment | Area                                         | Condition    | Organism |
|-------------------------------------|--------|-------------------|----------------------------------------------|--------------|----------|
| *P. sylvestris*                     | Summer | Rhizosphere       | Sierra Nevada                                | Symptomatic  | Bacteria |
| *P. pinaster*                       | Spring | Root endosphere   | Sierra de Baza                               | Asymptomatic | Fungi    |
| *P. sylvestris* subsp. *nevadensis* |        |                   | Sierra de Almijara, Tejeda y Alhama          |              |          |


Here, "*Condition*" refers to the presence (or absence) of pine decline symptoms

12 tree replicates were selected in all cases, however, the development of the symptoms depended on the considered mountainous area and plant species. On the other hand, not all the plant host species were found in all the areas. Have a look at the following table to know the distribution of the samples:

| Area                                  | Species                             | Condition (Acronym)    |
|---------------------------------------|-------------------------------------|------------------------|
| Sierra Nevada                         | *P. sylvestris*                     | Symptomatic (NSD)      |
|                                       |                                     | Asymptomatic (NSH)     |
|                                       | *P. pinaster*                       | Symptomatic (NPD)      |
|                                       |                                     | Asymptomatic (NPH)     |
|                                       | *P. sylvestris* subsp. *nevadensis* | Symptomatic (NAD)      |
|                                       |                                     | Asymptomatic (NAH)     |
| Sierra de Baza                        | *P. sylvestris*                     | Symptomatic (BSD)      |
|                                       | *P. pinaster*                       | Symptomatic (BPD)      |
|                                       | *P. sylvestris* subsp. *nevadensis* | Asymptomatic (BAH)     |
| Sierra de Almijara, Tejeda y Alhama   | *P. sylvestris*                     | Asymptomatic (ASH)     |
|                                       | *P. pinaster*                       | Asymptomatic (APH)     |


Thus, we have 11 groups of samples x 2 Seasons x 2 Plant compartments x 2 Organisms x 12 trees per group of samples = 1056 samples.

# What will I find in this GitHub project?

This repository contains different scripts: those related to the DNA sequencing for fungal and bacterial datasets, and other related to the preparation and standardization of the data to the DarwinCore Archive to integrate the data into GBIF

## Sequencing data
These scripts explain all the steps from the fastq files to a filtered ASV table.

- [bacterial](sequencing_reads_processing_16S.qmd)
- [fungal](sequencing_reads_processing_ITS2.qmd)

## Data analysis 
All the steps from the previously obtained ASV table to the obtaining of the rarefaction curves are explained.

- [bacterial](analysis_16S.qmd)
- [fungal](analysis_ITS2.qmd)

## Data integration into GBIF
- [Steps to accommodate the data into Darwin-Core Archive (for GBIF Integration)]()

# Processing of Illumina MiSeq sequencing reads (16S rRNA V3-V4 hypervariable regions and ITS2)
## Sample sequencing scheme
Total DNA was extracted from rhizosphere and root endosphere samples according to @Lasa2024. Due to the high number of samples, DNA had to be sequenced in different MiSeq runs. So, finally, we got the following sequencing scheme:

 - Bacteria + Fungal samples:
	 - Run 1: Samples from rhizosphere, spring 2022.
	 - Run 2: Samples from endosphere, spring 2022.
	 - Run 3: Samples from rhizosphere, summer 2021.
	 - Run 4: samples from endosphere, summer 2021.
	 - Run 5: some samples from endosphere and other samples from root endosphere, summer 2021.
It should be taken into account that in each sequencing run, 3 replicates of a MockCommunity were included.

The Data descriptor manuscript describes the data corresponding to the sub-project that included just ***P. sylvestris*** trees from **Sierra Nevada, Sierra de Baza and Sierra de Almijara, Tejeda y Alhama** (both symptomatic and decline-affected trees, according to the Table above) obtained in **spring 2022**. In this project, both **fungal** and **bacterial** communities inhabiting the **rhizosphere** and **root endosphere** of the selected trees were considered.

## Sample sequencing scheme
According to DADA2 developers, when our scientific projects include 2 or more sequencing runs, they should be processed separately. Once the Sequence Tables are obtained, all of them have to be merged into a one merged table, and we can go on processing all the samples of the same scientific project from this unique merged table.
In our case, the 528 bacterial samples (+replicates of the MockCommunity) were thus separately processed according to the corresponding runs. Then, the 5 Sequence Table were joined into a just one table, and we followed processing that table. The same procedure was applied for fungal dataset.

Thus, all the samples of SUMHAL project were further processed altogether, although the Data Descriptor **just** included samples corresponding to *P. sylvestris* trees (spring 2022) located in the three selected areas. Accordingly, the BioProject PRJNA993625 (NCBI SRA) included just the fastq files corresponding to this sub-project. In a near future, a second BioProject will be created again in NCBI SRA including the fastq files corresponding to another sub-projects of the SUMHAL project


## Authors
- [Ana V. Lasa](https://github.com/anitalasa) <a href="https://orcid.org/0000-0003-3783-7157" target="orcid.widget">
<img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /></a>

- [Antonio J. Pérez-Luque](https://github.com/ajpelu) <a href="https://orcid.org/0000-0002-1747-0469" target="orcid.widget">
<img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /></a>

- Antonio José Fernández-González <a href="https://orcid.org/0000-0002-1938-5436" target="orcid.widget">
<img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /></a>

- Pablo J. Villadas <a href="https://orcid.org/0000-0001-8662-8883" target="orcid.widget">
<img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /></a>

- Jesús Mercado-Blanco <a href="https://orcid.org/0000-0003-1895-5895" target="orcid.widget">
<img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /></a>

- Manuel Fernández-López <a href="https://orcid.org/0000-0003-2366-6908" target="orcid.widget">
<img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /></a>

## How to cite: 

### Repository: 
Ana V. Lasa & Antonio J. Pérez-Luque (2024) Code repository of the DataPaper: Root-associated microbiota of decline-affected and asymptomatic *Pinus sylvestris* trees.  https://doi.org/10.5281/zenodo.13711241

### Main paper: 
Lasa, A.V.; Fernández-González, A.J.; Villadas, P.J.; Mercado-Blanco, J.; Pérez-Luque, A.J.; Fernández-López, M. (2024). Mediterranean pine forest decline: a matter of root-associated microbiota and climate change. Science of The Total Environment 926: 171858. https://doi.org/10.1016/j.scitotenv.2024.171858

### Data Paper: 
Building, still unpublished

# References 
