<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.552">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Script for data paper: Root-associated microbiota of decline-affected and asymptomatic Pinus sylvestris trees</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link active" href="./index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bacteria" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Bacteria</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-bacteria">    
        <li>
    <a class="dropdown-item" href="./analysis_16S.html">
 <span class="dropdown-text">Data Analysis 16S</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./sequencing_reads_processing_16S.html">
 <span class="dropdown-text">Sequencing Reads Processing 16</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-fungi" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Fungi</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-fungi">    
        <li>
    <a class="dropdown-item" href="./analysis_ITS2.html">
 <span class="dropdown-text">Data Analysis ITS2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./sequencing_reads_processing_ITS2.html">
 <span class="dropdown-text">Sequencing Reads Processing ITS2</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="./prepare_dwc.html"> 
<span class="menu-text">DarwinCore accomodation</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a></li>
  <li><a href="#experimental-design" id="toc-experimental-design" class="nav-link" data-scroll-target="#experimental-design">Experimental design</a></li>
  <li><a href="#what-will-i-find-in-this-github-project" id="toc-what-will-i-find-in-this-github-project" class="nav-link" data-scroll-target="#what-will-i-find-in-this-github-project">What will I find in this GitHub project?</a>
  <ul class="collapse">
  <li><a href="#sequencing-data" id="toc-sequencing-data" class="nav-link" data-scroll-target="#sequencing-data">Sequencing data</a></li>
  <li><a href="#data-analysis" id="toc-data-analysis" class="nav-link" data-scroll-target="#data-analysis">Data analysis</a></li>
  <li><a href="#data-integration-into-gbif" id="toc-data-integration-into-gbif" class="nav-link" data-scroll-target="#data-integration-into-gbif">Data integration into GBIF</a></li>
  </ul></li>
  <li><a href="#processing-of-illumina-miseq-sequencing-reads-16s-v3-v4-and-its2" id="toc-processing-of-illumina-miseq-sequencing-reads-16s-v3-v4-and-its2" class="nav-link" data-scroll-target="#processing-of-illumina-miseq-sequencing-reads-16s-v3-v4-and-its2">Processing of Illumina MiSeq sequencing reads (16S V3-V4 and ITS2)</a>
  <ul class="collapse">
  <li><a href="#sample-sequencing-scheme" id="toc-sample-sequencing-scheme" class="nav-link" data-scroll-target="#sample-sequencing-scheme">Sample sequencing scheme</a></li>
  <li><a href="#sample-sequencing-scheme-1" id="toc-sample-sequencing-scheme-1" class="nav-link" data-scroll-target="#sample-sequencing-scheme-1">Sample sequencing scheme</a></li>
  <li><a href="#authors" id="toc-authors" class="nav-link" data-scroll-target="#authors">Authors</a></li>
  <li><a href="#how-to-cite" id="toc-how-to-cite" class="nav-link" data-scroll-target="#how-to-cite">How to cite:</a>
  <ul class="collapse">
  <li><a href="#repository" id="toc-repository" class="nav-link" data-scroll-target="#repository">Repository:</a></li>
  <li><a href="#main-paper" id="toc-main-paper" class="nav-link" data-scroll-target="#main-paper">Main paper:</a></li>
  <li><a href="#data-paper" id="toc-data-paper" class="nav-link" data-scroll-target="#data-paper">Data Paper:</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Script for data paper: Root-associated microbiota of decline-affected and asymptomatic <em>Pinus sylvestris</em> trees</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>This repository contains scripts and analysis used for the datapaper “Root-associated microbiota of decline-affected and asymptomatic <em>Pinus sylvestris</em> trees”</p>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>The SUMHAL Project aims at deciphering the changes in the microbial communities associated to the roots of different <em>Pinus</em> species as a consequence of forest decline. Three different mountainous areas covered by <em>Pinus</em> spp. stands in the Southeast Spain were selected for this purpose. Since different pine species in different conditions were selected, different sub-project were created, which resulted in different scientific publications. Below you will find specific information about the main project and sub-projects</p>
</section>
<section id="experimental-design" class="level1">
<h1>Experimental design</h1>
<p>In the following table, all the information regarding the project is summarized</p>
<table class="table">
<colgroup>
<col style="width: 27%">
<col style="width: 5%">
<col style="width: 14%">
<col style="width: 34%">
<col style="width: 10%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th>Species</th>
<th>Season</th>
<th>Plant compartment</th>
<th>Area</th>
<th>Condition</th>
<th>Organism</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><em>P. sylvestris</em></td>
<td>Summer</td>
<td>Rhizosphere</td>
<td>Sierra Nevada</td>
<td>Symptomatic</td>
<td>Bacteria</td>
</tr>
<tr class="even">
<td><em>P. pinaster</em></td>
<td>Spring</td>
<td>Root endosphere</td>
<td>Sierra de Baza</td>
<td>Asymptomatic</td>
<td>Fungi</td>
</tr>
<tr class="odd">
<td><em>P. sylvestris</em> subsp. <em>nevadensis</em></td>
<td></td>
<td></td>
<td>Sierra de Almijara, Tejeda y Alhama</td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<p>Here, “<em>Condition</em>” refers to the presence (or absence) of pine decline symptoms</p>
<p>12 tree replicates were selected in all cases, however, the development of the symptoms depended on the considered mountainous area and plant species. On the other hand, not all the plant host species were found in all the areas. Have a look at the following table to know the distribution of the samples:</p>
<table class="table">
<colgroup>
<col style="width: 39%">
<col style="width: 37%">
<col style="width: 24%">
</colgroup>
<thead>
<tr class="header">
<th>Area</th>
<th>Species</th>
<th>Condition (Acronym)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Sierra Nevada</td>
<td><em>P. sylvestris</em></td>
<td>Symptomatic (NSD)</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>Asymptomatic (NSH)</td>
</tr>
<tr class="odd">
<td></td>
<td><em>P. pinaster</em></td>
<td>Symptomatic (NPD)</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>Asymptomatic (NPH)</td>
</tr>
<tr class="odd">
<td></td>
<td><em>P. sylvestris</em> subsp. <em>nevadensis</em></td>
<td>Symptomatic (NAD)</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td>Asymptomatic (NAH)</td>
</tr>
<tr class="odd">
<td>Sierra de Baza</td>
<td><em>P. sylvestris</em></td>
<td>Symptomatic (BSD)</td>
</tr>
<tr class="even">
<td></td>
<td><em>P. pinaster</em></td>
<td>Symptomatic (BPD)</td>
</tr>
<tr class="odd">
<td></td>
<td><em>P. sylvestris</em> subsp. <em>nevadensis</em></td>
<td>Asymptomatic (BAH)</td>
</tr>
<tr class="even">
<td>Sierra de Almijara, Tejeda y Alhama</td>
<td><em>P. sylvestris</em></td>
<td>Asymptomatic (ASH)</td>
</tr>
<tr class="odd">
<td></td>
<td><em>P. pinaster</em></td>
<td>Asymptomatic (APH)</td>
</tr>
</tbody>
</table>
<p>Thus, we have 11 groups of samples x 2 Seasons x 2 Plant compartments x 2 Organisms x 12 trees per group of samples = 1056 samples.</p>
</section>
<section id="what-will-i-find-in-this-github-project" class="level1">
<h1>What will I find in this GitHub project?</h1>
<p>This repository contains different scripts: those related to the DNA sequencing for fungal an bacterial datasets, and other related to the preparation and standardization of the data to the DarwinCore Archive to integtrate the data into GBIF</p>
<section id="sequencing-data" class="level2">
<h2 class="anchored" data-anchor-id="sequencing-data">Sequencing data</h2>
<p>This scripts explaing all the steps the steps from the fastq files to a filtered ASV table.</p>
<ul>
<li><a href="./sequencing_reads_processing_16S.html">bacterial</a></li>
<li><a href="./sequencing_reads_processing_ITS2.html">fungal</a></li>
</ul>
</section>
<section id="data-analysis" class="level2">
<h2 class="anchored" data-anchor-id="data-analysis">Data analysis</h2>
<p>All the steps from the previously obtained ASV table to the obtaining of the rarefaction curves are explained.</p>
<ul>
<li><a href="./analysis_16S.html">bacterial</a></li>
<li><a href="./analysis_ITS2.html">fungal</a></li>
</ul>
</section>
<section id="data-integration-into-gbif" class="level2">
<h2 class="anchored" data-anchor-id="data-integration-into-gbif">Data integration into GBIF</h2>
<ul>
<li><a href="">Steps to accommodate the data into Darwin-Core Archive (for GBIF Integration)</a></li>
</ul>
</section>
</section>
<section id="processing-of-illumina-miseq-sequencing-reads-16s-v3-v4-and-its2" class="level1">
<h1>Processing of Illumina MiSeq sequencing reads (16S V3-V4 and ITS2)</h1>
<section id="sample-sequencing-scheme" class="level2">
<h2 class="anchored" data-anchor-id="sample-sequencing-scheme">Sample sequencing scheme</h2>
<p>Total DNA was extracted from rhizosphere and root endosphere samples according to <span class="citation" data-cites="Lasa2024">Lasa et al. (<a href="#ref-Lasa2024" role="doc-biblioref">2024</a>)</span>. Due to the high number of samples, DNA had to be sequenced in different MiSeq runs. So, finally, we got the following sequencing scheme:</p>
<ul>
<li>Bacteria + Fungal samples:
<ul>
<li>Run 1: Samples from rhizosphere, spring 2022.</li>
<li>Run 2: Samples from endosphere, spring 2022.</li>
<li>Run 3: Samples from rhizosphere, summer 2021.</li>
<li>Run 4: samples from endosphere, summer 2021.</li>
<li>Run 5: some samples from endosphere and other samples from root endosphere, summer 2021. It should be taken into account that in each sequencing run, 3 replicates of a MockCommunity were included.</li>
</ul></li>
</ul>
<p>The Data descriptor manuscript describes the data corresponding to the sub-project that included just <strong><em>P. sylvestris</em></strong> trees from <strong>Sierra Nevada, Sierra de Baza and Sierra de Almijara, Tejeda y Alhama</strong> (both symptomatic and decline-affected trees, according the Table above) obtained in <strong>spring 2022</strong>. In this project, both <strong>fungal</strong> and <strong>bacterial</strong> communities inhabiting the <strong>rhizosphere</strong> and <strong>root endosphere</strong> of the selected trees were considered.</p>
</section>
<section id="sample-sequencing-scheme-1" class="level2">
<h2 class="anchored" data-anchor-id="sample-sequencing-scheme-1">Sample sequencing scheme</h2>
<p>According to DADA2 developers, when our scientific projects include 2 or more sequencing runs, they should be processed separatedly. Once the Sequence Tables are obtained, all of them have to be merged into a one merged table, and we can go on processing all the samples of the same scientific project from this unique merged table. In our case, the 528 bacterial samples (+replicates of the MockCommunity) were thus separatedly processed according to the corresponding runs. Then, the 5 Sequence Table were joined into a just one table, all we followed processing that table. The same procedure was applied for fungal dataset.</p>
<p>Thus, all the samples of SUMHAL project were futher processed altogether, although the Data Descriptor <strong>just</strong> included samples corresponding to <em>P. sylvestris</em> trees (spring 2022) located in the three selected areas. Accordingly, the BioProject PRJNA993625 (NCBI SRA) included just the fastq files corresponding to this sub-project. In a near future, a second BioProject will be created again in NCBI SRA including the fastq files corresponding to another sub-project of the SUMHAL project</p>
</section>
<section id="authors" class="level2">
<h2 class="anchored" data-anchor-id="authors">Authors</h2>
<ul>
<li><p>Ana V Lasa <a href="https://orcid.org/0000-0003-3783-7157" target="orcid.widget"> <img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16"></a></p></li>
<li><p><a href="https://github.com/ajpelu">Antonio J. Pérez-Luque</a> <a href="https://orcid.org/0000-0002-1747-0469" target="orcid.widget"> <img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16"></a></p></li>
<li><p>Antonio José Fernández-González <a href="https://orcid.org/0000-0002-1938-5436" target="orcid.widget"> <img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16"></a></p></li>
<li><p>Pablo J. Villadas <a href="https://orcid.org/0000-0001-8662-8883" target="orcid.widget"> <img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16"></a></p></li>
<li><p>Jesús Mercado-Blanco <a href="https://orcid.org/0000-0003-1895-5895" target="orcid.widget"> <img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16"></a></p></li>
<li><p>Manuel Fernández-López <a href="https://orcid.org/0000-0003-2366-6908" target="orcid.widget"> <img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16"></a></p></li>
</ul>
</section>
<section id="how-to-cite" class="level2">
<h2 class="anchored" data-anchor-id="how-to-cite">How to cite:</h2>
<section id="repository" class="level3">
<h3 class="anchored" data-anchor-id="repository">Repository:</h3>
<p>Ana V. Lasa &amp; Antonio J. Pérez-Luque (2024) Code repository of the DataPaper: Root-associated microbiota of decline-affected and asymptomatic <em>Pinus sylvestris</em> trees. INCLUIR DOI ZENODO https://doi.org/10.5281/zenodo.7350714</p>
</section>
<section id="main-paper" class="level3">
<h3 class="anchored" data-anchor-id="main-paper">Main paper:</h3>
<p>Lasa, A.V.; Fernández-González, A.J.; Villadas, P.J.; Mercado-Blanco, J.; Pérez-Luque, A.J.; Fernández-López, M. (2024). Mediterranean pine forest decline: a matter of root-associated microbiota and climate change. Science of The Total Environment 926: 171858. https://doi.org/10.1016/j.scitotenv.2024.171858</p>
</section>
<section id="data-paper" class="level3">
<h3 class="anchored" data-anchor-id="data-paper">Data Paper:</h3>
<p>Include the cite when the DataPaper is published</p>
</section>
</section>
</section>
<section id="references" class="level1">



<!-- -->


</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-Lasa2024" class="csl-entry" role="listitem">
Lasa, Ana V., Antonio José Fernández-González, Pablo J. Villadas, Jesús Mercado-Blanco, Antonio J. Pérez-Luque, and Manuel Fernández-López. 2024. <span>“Mediterranean Pine Forest Decline: A Matter of Root-Associated Microbiota and Climate Change.”</span> <em>Science of The Total Environment</em> 926: 171858. https://doi.org/<a href="https://doi.org/10.1016/j.scitotenv.2024.171858">https://doi.org/10.1016/j.scitotenv.2024.171858</a>.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb1" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Script for data paper: Root-associated microbiota of decline-affected and asymptomatic *Pinus sylvestris* trees"</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> references.bib</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>This repository contains scripts and analysis used for the datapaper "Root-associated microbiota of decline-affected and asymptomatic *Pinus sylvestris* trees" </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction  </span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>The SUMHAL Project aims at deciphering the changes in the microbial communities associated to the roots of different *Pinus* species as a consequence of forest decline. Three different mountainous areas covered by *Pinus* spp. stands in the Southeast Spain were selected for this purpose.</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>Since different pine species in different conditions were selected, different sub-project were created, which resulted in different scientific publications. Below you will find specific information about the main project and sub-projects</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu"># Experimental design</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>In the following table, all the information regarding the project is summarized </span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>| Species                             | Season | Plant compartment | Area                                         | Condition    | Organism |</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>|-------------------------------------|--------|-------------------|----------------------------------------------|--------------|----------|</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>| *P. sylvestris*                     | Summer | Rhizosphere       | Sierra Nevada                                | Symptomatic  | Bacteria |</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>| *P. pinaster*                       | Spring | Root endosphere   | Sierra de Baza                               | Asymptomatic | Fungi    |</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>| *P. sylvestris* subsp. *nevadensis* |        |                   | Sierra de Almijara, Tejeda y Alhama          |              |          |</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>Here, "*Condition*" refers to the presence (or absence) of pine decline symptoms</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>12 tree replicates were selected in all cases, however, the development of the symptoms depended on the considered mountainous area and plant species. On the other hand, not all the plant host species were found in all the areas. Have a look at the following table to know the distribution of the samples:</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>| Area                                  | Species                             | Condition (Acronym)    |</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>|---------------------------------------|-------------------------------------|------------------------|</span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>| Sierra Nevada                         | *P. sylvestris*                     | Symptomatic (NSD)      |</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>|                                       |                                     | Asymptomatic (NSH)     |</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>|                                       | *P. pinaster*                       | Symptomatic (NPD)      |</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>|                                       |                                     | Asymptomatic (NPH)     |</span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>|                                       | *P. sylvestris* subsp. *nevadensis* | Symptomatic (NAD)      |</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>|                                       |                                     | Asymptomatic (NAH)     |</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>| Sierra de Baza                        | *P. sylvestris*                     | Symptomatic (BSD)      |</span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>|                                       | *P. pinaster*                       | Symptomatic (BPD)      |</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>|                                       | *P. sylvestris* subsp. *nevadensis* | Asymptomatic (BAH)     |</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>| Sierra de Almijara, Tejeda y Alhama   | *P. sylvestris*                     | Asymptomatic (ASH)     |</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>|                                       | *P. pinaster*                       | Asymptomatic (APH)     |</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>Thus, we have 11 groups of samples x 2 Seasons x 2 Plant compartments x 2 Organisms x 12 trees per group of samples = 1056 samples.</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="fu"># What will I find in this GitHub project?</span></span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>This repository contains different scripts: those related to the DNA sequencing for fungal an bacterial datasets, and other related to the preparation and standardization of the data to the DarwinCore Archive to integtrate the data into GBIF</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a><span class="fu">## Sequencing data</span></span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>This scripts explaing all the steps the steps from the fastq files to a filtered ASV table.</span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">bacterial</span><span class="co">](sequencing_reads_processing_16S.qmd)</span></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">fungal</span><span class="co">](sequencing_reads_processing_ITS2.qmd)</span></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data analysis </span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>All the steps from the previously obtained ASV table to the obtaining of the rarefaction curves are explained.</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">bacterial</span><span class="co">](analysis_16S.qmd)</span></span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">fungal</span><span class="co">](analysis_ITS2.qmd)</span></span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a><span class="fu">## Data integration into GBIF</span></span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Steps to accommodate the data into Darwin-Core Archive (for GBIF Integration)</span><span class="co">]()</span></span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a><span class="fu"># Processing of Illumina MiSeq sequencing reads (16S V3-V4 and ITS2)</span></span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a><span class="fu">## Sample sequencing scheme</span></span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>Total DNA was extracted from rhizosphere and root endosphere samples according to @Lasa2024. Due to the high number of samples, DNA had to be sequenced in different MiSeq runs. So, finally, we got the following sequencing scheme:</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a><span class="ss"> - </span>Bacteria + Fungal samples:</span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a><span class="ss">     - </span>Run 1: Samples from rhizosphere, spring 2022.</span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a><span class="ss">     - </span>Run 2: Samples from endosphere, spring 2022.</span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a><span class="ss">     - </span>Run 3: Samples from rhizosphere, summer 2021.</span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a><span class="ss">     - </span>Run 4: samples from endosphere, summer 2021.</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a><span class="ss">     - </span>Run 5: some samples from endosphere and other samples from root endosphere, summer 2021.</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a>It should be taken into account that in each sequencing run, 3 replicates of a MockCommunity were included.</span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a>The Data descriptor manuscript describes the data corresponding to the sub-project that included just ***P. sylvestris*** trees from **Sierra Nevada, Sierra de Baza and Sierra de Almijara, Tejeda y Alhama** (both symptomatic and decline-affected trees, according the Table above) obtained in **spring 2022**. In this project, both **fungal** and **bacterial** communities inhabiting the **rhizosphere** and **root endosphere** of the selected trees were considered.</span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a><span class="fu">## Sample sequencing scheme</span></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a>According to DADA2 developers, when our scientific projects include 2 or more sequencing runs, they should be processed separatedly. Once the Sequence Tables are obtained, all of them have to be merged into a one merged table, and we can go on processing all the samples of the same scientific project from this unique merged table.</span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a>In our case, the 528 bacterial samples (+replicates of the MockCommunity) were thus separatedly processed according to the corresponding runs. Then, the 5 Sequence Table were joined into a just one table, all we followed processing that table. The same procedure was applied for fungal dataset.</span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a>Thus, all the samples of SUMHAL project were futher processed altogether, although the Data Descriptor **just** included samples corresponding to *P. sylvestris* trees (spring 2022) located in the three selected areas. Accordingly, the BioProject PRJNA993625 (NCBI SRA) included just the fastq files corresponding to this sub-project. In a near future, a second BioProject will be created again in NCBI SRA including the fastq files corresponding to another sub-project of the SUMHAL project</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a><span class="fu">## Authors</span></span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Ana V Lasa  &lt;a href="https://orcid.org/0000-0003-3783-7157" target="orcid.widget"&gt;</span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a>&lt;img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /&gt;&lt;/a&gt;</span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span><span class="co">[</span><span class="ot">Antonio J. Pérez-Luque</span><span class="co">](https://github.com/ajpelu)</span> &lt;a href="https://orcid.org/0000-0002-1747-0469" target="orcid.widget"&gt;</span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a>&lt;img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /&gt;&lt;/a&gt;</span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Antonio José Fernández-González &lt;a href="https://orcid.org/0000-0002-1938-5436" target="orcid.widget"&gt;</span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>&lt;img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /&gt;&lt;/a&gt;</span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Pablo J. Villadas &lt;a href="https://orcid.org/0000-0001-8662-8883" target="orcid.widget"&gt;</span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>&lt;img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /&gt;&lt;/a&gt;</span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Jesús Mercado-Blanco &lt;a href="https://orcid.org/0000-0003-1895-5895" target="orcid.widget"&gt;</span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a>&lt;img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /&gt;&lt;/a&gt;</span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Manuel Fernández-López &lt;a href="https://orcid.org/0000-0003-2366-6908" target="orcid.widget"&gt;</span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>&lt;img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /&gt;&lt;/a&gt;</span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a><span class="fu">## How to cite: </span></span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a><span class="fu">### Repository: </span></span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>Ana V. Lasa &amp; Antonio J. Pérez-Luque (2024) Code repository of the DataPaper: Root-associated microbiota of decline-affected and asymptomatic *Pinus sylvestris* trees. INCLUIR DOI ZENODO https://doi.org/10.5281/zenodo.7350714</span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-109"><a href="#cb1-109" aria-hidden="true" tabindex="-1"></a><span class="fu">### Main paper: </span></span>
<span id="cb1-110"><a href="#cb1-110" aria-hidden="true" tabindex="-1"></a>Lasa, A.V.; Fernández-González, A.J.; Villadas, P.J.; Mercado-Blanco, J.; Pérez-Luque, A.J.; Fernández-López, M. (2024). Mediterranean pine forest decline: a matter of root-associated microbiota and climate change. Science of The Total Environment 926: 171858. https://doi.org/10.1016/j.scitotenv.2024.171858 </span>
<span id="cb1-111"><a href="#cb1-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-112"><a href="#cb1-112" aria-hidden="true" tabindex="-1"></a><span class="fu">### Data Paper: </span></span>
<span id="cb1-113"><a href="#cb1-113" aria-hidden="true" tabindex="-1"></a>Include the cite when the DataPaper is published </span>
<span id="cb1-114"><a href="#cb1-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-115"><a href="#cb1-115" aria-hidden="true" tabindex="-1"></a><span class="fu"># References </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>