<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.552">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Ana V Lasa  ">

<title>Sequencing Reads Processing 16S</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-bacteria" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Bacteria</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-bacteria">    
        <li>
    <a class="dropdown-item" href="./analysis_16S.html">
 <span class="dropdown-text">Data Analysis 16S</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./sequencing_reads_processing_16S.html">
 <span class="dropdown-text">16S sequencing reads processing</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item dropdown ">
    <a class="nav-link dropdown-toggle" href="#" id="nav-menu-fungi" role="button" data-bs-toggle="dropdown" aria-expanded="false">
 <span class="menu-text">Fungi</span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="nav-menu-fungi">    
        <li>
    <a class="dropdown-item" href="./analysis_ITS2.html">
 <span class="dropdown-text">Data Analysis ITS2</span></a>
  </li>  
        <li>
    <a class="dropdown-item" href="./sequencing_reads_processing_ITS2.html">
 <span class="dropdown-text">ITS2 sequencing reads processing</span></a>
  </li>  
    </ul>
  </li>
  <li class="nav-item">
    <a class="nav-link" href="./prepare_dwc.html"> 
<span class="menu-text">DarwinCore accomodation</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#aim" id="toc-aim" class="nav-link active" data-scroll-target="#aim">Aim</a></li>
  <li><a href="#step-1-adaption-of-the-file-names" id="toc-step-1-adaption-of-the-file-names" class="nav-link" data-scroll-target="#step-1-adaption-of-the-file-names">Step 1: Adaption of the file names</a></li>
  <li><a href="#step-2-check-the-quality-of-the-reads" id="toc-step-2-check-the-quality-of-the-reads" class="nav-link" data-scroll-target="#step-2-check-the-quality-of-the-reads">Step 2: Check the quality of the reads</a></li>
  <li><a href="#step-3.-figaro" id="toc-step-3.-figaro" class="nav-link" data-scroll-target="#step-3.-figaro">Step 3. FIGARO</a></li>
  <li><a href="#step-4.-filter-and-trimming-steps" id="toc-step-4.-filter-and-trimming-steps" class="nav-link" data-scroll-target="#step-4.-filter-and-trimming-steps">Step 4. Filter and trimming steps</a></li>
  <li><a href="#step-5-cutadapt.-primer-removal-from-f-and-r-reads" id="toc-step-5-cutadapt.-primer-removal-from-f-and-r-reads" class="nav-link" data-scroll-target="#step-5-cutadapt.-primer-removal-from-f-and-r-reads">Step 5: CUTADAPT. primer removal from F and R reads</a></li>
  <li><a href="#step-6-dada2" id="toc-step-6-dada2" class="nav-link" data-scroll-target="#step-6-dada2">Step 6: DADA2</a></li>
  <li><a href="#step-7-merge-seqtabs-obtained-from-different-sequencing-runs" id="toc-step-7-merge-seqtabs-obtained-from-different-sequencing-runs" class="nav-link" data-scroll-target="#step-7-merge-seqtabs-obtained-from-different-sequencing-runs">Step 7: Merge SeqTabs obtained from different sequencing runs</a></li>
  <li><a href="#step-8-chimera-removal" id="toc-step-8-chimera-removal" class="nav-link" data-scroll-target="#step-8-chimera-removal">Step 8: Chimera removal</a></li>
  <li><a href="#step-9-check-the-number-of-sequences-retained-in-each-step-of-the-processing" id="toc-step-9-check-the-number-of-sequences-retained-in-each-step-of-the-processing" class="nav-link" data-scroll-target="#step-9-check-the-number-of-sequences-retained-in-each-step-of-the-processing">Step 9: Check the number of sequences retained in each step of the processing</a></li>
  <li><a href="#step-10-taxonomical-classification" id="toc-step-10-taxonomical-classification" class="nav-link" data-scroll-target="#step-10-taxonomical-classification">Step 10: Taxonomical classification</a></li>
  <li><a href="#step-11-obtaining-the-asv-table" id="toc-step-11-obtaining-the-asv-table" class="nav-link" data-scroll-target="#step-11-obtaining-the-asv-table">Step 11: Obtaining the ASV table</a></li>
  <li><a href="#step-12-mock-community-quality-control" id="toc-step-12-mock-community-quality-control" class="nav-link" data-scroll-target="#step-12-mock-community-quality-control">Step 12: Mock community quality control</a></li>
  <li><a href="#step-13-removal-of-plastids-plants-chloroplast-and-other-artifcats-sequences" id="toc-step-13-removal-of-plastids-plants-chloroplast-and-other-artifcats-sequences" class="nav-link" data-scroll-target="#step-13-removal-of-plastids-plants-chloroplast-and-other-artifcats-sequences">Step 13: Removal of Plastids, Plants, Chloroplast and other artifcats sequences</a></li>
  <li><a href="#step-14-refinement-of-the-taxonomy" id="toc-step-14-refinement-of-the-taxonomy" class="nav-link" data-scroll-target="#step-14-refinement-of-the-taxonomy">Step 14: Refinement of the taxonomy</a></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Sequencing Reads Processing 16S</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Ana V Lasa <a href="https://orcid.org/0000-0003-3783-7157" target="orcid.widget"> <img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16"></a> </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="aim" class="level1 unnumbered">
<h1 class="unnumbered">Aim</h1>
<p>Processing of Illumina (MiSeq reads). 16S (bacteria) reads</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"dada2"</span>)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="co"># phyloseq</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"phyloseq"</span>)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Biocgenerics</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"BiocGenerics"</span>)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Biostrings</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"Biostrings"</span>)</span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a><span class="co"># ShortRead</span></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"ShortRead"</span>)</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot2 (for plotting)</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a><span class="co"># tidyverse</span></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Load corresponding libraries</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dada2)</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiocGenerics)</span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ShortRead)</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="st">"C:/Users/radik/Desktop/Psylvestris_DataPper"</span> <span class="co">#set the path where the fastq.gz files are</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>(path)<span class="co">#check that all the needed files are in the path</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co">#NOTE: if our scientific project includes different sequencing runs (for instance, if you have a lot of samples that do not fit well into just one run), </span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co">#we should repeat steps 1) to 6) for each sequencing run, and then (step 7), merge all the sequence tables obtained in the step 6)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>If our scientific project includes different sequencing runs (for instance, if you have a lot of samples that do not fit well into just one run), we should repeat steps 1) to 6) for each sequencing run, and then (step 7), merge all the sequence tables obtained in the step 6).</p>
</section>
<section id="step-1-adaption-of-the-file-names" class="level1">
<h1>Step 1: Adaption of the file names</h1>
<ul>
<li>Sort Forward (F) and Reverse (R) files separatedly</li>
<li>F and R fastq filenames should have the following format: SAMPLENAME_R1_001.fastq.gz;SAMPLENAME_R2_001.fastq.gz</li>
<li>In this example, the names of F and R file are the following: NGS006-22-EXP4-ITS2-SAMPLENAME_yyy-R1_001.fastq.gz</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>fnFs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">"_R1_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>fnRs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">"_R2_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract sample names, assuming filenames have format: xxx-SAMPLENAME_yyy_R1_001.fastq.gz</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>sample.names_raw <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">basename</span>(fnFs), <span class="st">"_"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co"># this function splits the string when it finds the first "_" symbol, and keeps the first ("1") part of the string</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># output in this example: NGS006-22-EXP5-16S-SAMPLENAME</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>sample.names <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"NGS006-22-EXP5-16S-"</span>, <span class="st">""</span>, sample.names_raw) <span class="co"># replace the xxx part of the string, by nothing (or by the characters that allow you to get the sample names</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co"># output in this example: SAMPLENAME</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-2-check-the-quality-of-the-reads" class="level1">
<h1>Step 2: Check the quality of the reads</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># a) Check the number of reads</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>raw_reads_count <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fnFs)) {</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  raw_reads_count <span class="ot">&lt;-</span> <span class="fu">rbind</span>(raw_reads_count, <span class="fu">length</span>(ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs[i])))</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>} <span class="co"># it counts the number of reads in fnFs</span></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw_reads_count) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">"_"</span> <span class="ot">=</span> <span class="fu">rownames</span>(raw_reads_count), raw_reads_count)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(a) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Sample"</span>, <span class="st">"Number_of_reads"</span>)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Check which samples have the highlest and lowest amount of reads</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.min</span>(raw_reads_count)], <span class="fu">min</span>(raw_reads_count))</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.max</span>(raw_reads_count)], <span class="fu">max</span>(raw_reads_count))</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(<span class="st">"_"</span> <span class="ot">=</span> <span class="fu">rownames</span>(raw_reads_count), raw_reads_count), <span class="at">file =</span> <span class="st">"NumberRawReads.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> F)</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a><span class="co"># b) Check the length of the reads</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>reads <span class="ot">&lt;-</span> ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs) <span class="co"># it saves the reads into a new variable</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>uniques <span class="ot">&lt;-</span> <span class="fu">unique</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width) <span class="co"># to get the length of the reads (bp)</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="cn">NULL</span> <span class="co"># it counts the number of reads of each length</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(uniques)) {</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">rbind</span>(counts, <span class="fu">length</span>(<span class="fu">which</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width <span class="sc">==</span> uniques[i])))</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>histogram <span class="ot">&lt;-</span> <span class="fu">cbind</span>(uniques, counts)</span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(histogram) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Seq.length"</span>, <span class="st">"counts"</span>)</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a><span class="co"># check the histogram</span></span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(histogram[<span class="fu">order</span>(histogram[, <span class="dv">1</span>], <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]) <span class="co"># Most sequences should fall in the expected sequence length</span></span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a><span class="co"># since we followed a 2x275 PE strategy, most of our reads are of 275-276bp</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(histogram, <span class="at">file =</span> <span class="st">"Length_Raw_Reads.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> F)</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the histogram</span></span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width, <span class="at">main =</span> <span class="st">"Forward length distribution"</span>, <span class="at">xlab =</span> <span class="st">"Sequence length"</span>, <span class="at">ylab =</span> <span class="st">"Raw reads"</span>)</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a><span class="co"># c) Check the quality plots</span></span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnFs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>]) <span class="co"># here you can plot whichever of the samples you want (in this example, samples number 4 and 5)</span></span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnRs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-3.-figaro" class="level1">
<h1>Step 3. FIGARO</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparation of the data</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Creamos una carpeta donde guarde todas las secuencias tras el corte por figaro</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>figFs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"figaro"</span>, <span class="fu">basename</span>(fnFs))</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>figRs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"figaro"</span>, <span class="fu">basename</span>(fnRs))</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Figaro requires all the reads are of the same length, so firstly we have to cut the sequences trying to</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a><span class="co"># include all the sequences, so we make an initial cut at 274 bp (we won't lose many reads)</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>out.figaro <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(fnFs, figFs, fnRs, figRs, <span class="at">compress =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>, <span class="at">truncLen =</span> <span class="fu">c</span>(<span class="dv">274</span>, <span class="dv">274</span>))</span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co"># filterAndTrim(input F, output F, input R, output R)</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a><span class="co"># IMPORTANT: figaro will consider these reads just for calculation of the best parameters for DADA2</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="co"># but these sequences will not be used for DADA2, we then will use all the reads for DADA2</span></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>FWD <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"CCTACGGGNBGCASCAG"</span>) <span class="co"># insert here the sequence of the the primer F</span></span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>REV <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GACTACNVGGGTATCTAATCC"</span>) <span class="co"># here the sequence of the primer R</span></span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a><span class="fu">nchar</span>(FWD) <span class="co"># to determine the number of bp of the primers F and R</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a><span class="fu">nchar</span>(REV)</span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Run figaro definitely:</span></span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>figaro <span class="ot">&lt;-</span> <span class="fu">system</span>((<span class="st">"python3 /home/programs/figaro/figaro/figaro.py -i C:/Users/radik/Desktop/Psylvestris_DataPper/figaro -o C:/Users/radik/Desktop/Psylvestris_DataPper/figaro -a 426 -f 17 -r 21"</span>),</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">intern =</span> <span class="cn">TRUE</span></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb5-23"><a href="#cb5-23" aria-hidden="true" tabindex="-1"></a><span class="co">#-i insert the path of the previously created figaro folder</span></span>
<span id="cb5-24"><a href="#cb5-24" aria-hidden="true" tabindex="-1"></a><span class="co">#-o the path of the figaro folder</span></span>
<span id="cb5-25"><a href="#cb5-25" aria-hidden="true" tabindex="-1"></a><span class="co">#-a amplicon size (bp) according to the 16S rRNA hypervariable regions under study</span></span>
<span id="cb5-26"><a href="#cb5-26" aria-hidden="true" tabindex="-1"></a><span class="co">#-f size of the F primer (bp)</span></span>
<span id="cb5-27"><a href="#cb5-27" aria-hidden="true" tabindex="-1"></a><span class="co">#-r size of the R primer (bp)</span></span>
<span id="cb5-28"><a href="#cb5-28" aria-hidden="true" tabindex="-1"></a><span class="co"># figaro script (figaro.py) runs properly in python3</span></span>
<span id="cb5-29"><a href="#cb5-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-30"><a href="#cb5-30" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(figaro) <span class="co"># check the results of figaro and select that maxExpectedError with which you obtain a good score and a good percentage of retained reads</span></span>
<span id="cb5-31"><a href="#cb5-31" aria-hidden="true" tabindex="-1"></a><span class="co"># write down the maxEE and the trimming positions, in this example: trimming positions [F,R]: [269,215]; maxEE[F,R]: [2,2]</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-4.-filter-and-trimming-steps" class="level1">
<h1>Step 4. Filter and trimming steps</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the folder "filtered" to save the trimmed reads</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>filtFs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnFs))</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>filtRs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnRs))</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="co"># trimming step itself:</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(fnFs, filtFs, fnRs, filtRs,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">truncLen =</span> <span class="fu">c</span>(<span class="dv">269</span>, <span class="dv">215</span>),</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">maxEE =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="at">truncQ =</span> <span class="dv">2</span>, <span class="at">rm.phix =</span> <span class="cn">TRUE</span>,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">compress =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>, <span class="at">minLen =</span> <span class="dv">50</span></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co"># input reads F, output reads F, input reads R, output reads R,</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="co"># truncLen: trimming positions in F and R reads, according to Figaro results</span></span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a><span class="co"># maxN: number of allowed ambiguities (N) in the reads</span></span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a><span class="co"># maxEE: maximum expected errors in F and R reads, according to Figaro results</span></span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a><span class="co"># trunqQ: the minimum quality score of each nucleotides; remove reads with at least one nucleotide with an associated quality score under the selected value</span></span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a><span class="co"># minLen: the minimum size of the reads; remove all the reads under this size (bp)</span></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(out)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-5-cutadapt.-primer-removal-from-f-and-r-reads" class="level1">
<h1>Step 5: CUTADAPT. primer removal from F and R reads</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function to calculate all the orientations of F and R primers</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">&lt;-</span> <span class="cf">function</span>(primer) {</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(Biostrings)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  dna <span class="ot">&lt;-</span> <span class="fu">DNAString</span>(primer) <span class="co"># BioStrings package works with strings but not with characters</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  orients <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">Forward =</span> dna, <span class="at">Complement =</span> Biostrings<span class="sc">::</span><span class="fu">complement</span>(dna), <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna),</span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna)</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString)) <span class="co"># change to string</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(FWD) <span class="co"># pass the function to primer F</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>REV.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(REV) <span class="co"># pass the function to primer R</span></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="co"># check the calculated orientations</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>REV.orients</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of times both primers (in all the possible orientations) appear in our dataset</span></span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">&lt;-</span> <span class="cf">function</span>(primer, fn) {</span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>  nhits <span class="ot">&lt;-</span> <span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a><span class="co"># check the number of reads in which the primers F and R (in all possible orientations) are found in sample number "1"</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">1</span>]]),</span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">1</span>]]),</span>
<span id="cb7-27"><a href="#cb7-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">1</span>]]),</span>
<span id="cb7-28"><a href="#cb7-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">1</span>]])</span>
<span id="cb7-29"><a href="#cb7-29" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-30"><a href="#cb7-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-31"><a href="#cb7-31" aria-hidden="true" tabindex="-1"></a><span class="co"># On the left, we have the presence of forward and reverse primers (FWD and REV) in forward and reverse reads (FWD.ForwardReads, FWD.ReverseReads, REV.ForwardReads and REV.ReverseReads). On the top of the table, we see the four possible orientations of the primers (Forward, Complement, Reverse and RevComp).</span></span>
<span id="cb7-32"><a href="#cb7-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-33"><a href="#cb7-33" aria-hidden="true" tabindex="-1"></a>cutadapt <span class="ot">&lt;-</span> <span class="st">"/usr/local/bin/cutadapt"</span> <span class="co"># path to cutadapt in your computer/machine</span></span>
<span id="cb7-34"><a href="#cb7-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-35"><a href="#cb7-35" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(<span class="st">"--version"</span>)) <span class="co"># Run shell commands from R</span></span>
<span id="cb7-36"><a href="#cb7-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-37"><a href="#cb7-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a directoy or folder where your reads will be saved</span></span>
<span id="cb7-38"><a href="#cb7-38" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"cutadapt"</span>)</span>
<span id="cb7-39"><a href="#cb7-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-40"><a href="#cb7-40" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb7-41"><a href="#cb7-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-42"><a href="#cb7-42" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnFs))</span>
<span id="cb7-43"><a href="#cb7-43" aria-hidden="true" tabindex="-1"></a>fnRs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnRs))</span>
<span id="cb7-44"><a href="#cb7-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-45"><a href="#cb7-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Produce arguments for cutadapt (visit the cutadapt website for more information)</span></span>
<span id="cb7-46"><a href="#cb7-46" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD) <span class="co"># calculate the reverse complementary sequence</span></span>
<span id="cb7-47"><a href="#cb7-47" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb7-48"><a href="#cb7-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-49"><a href="#cb7-49" aria-hidden="true" tabindex="-1"></a><span class="co"># add the specific adapters to F and R reads, which are need for cutadapt</span></span>
<span id="cb7-50"><a href="#cb7-50" aria-hidden="true" tabindex="-1"></a>R1.flags <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"-a"</span>, <span class="st">" "</span>, <span class="st">"^"</span>, FWD, <span class="st">"..."</span>, REV.RC)</span>
<span id="cb7-51"><a href="#cb7-51" aria-hidden="true" tabindex="-1"></a>R2.flags <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"-A"</span>, <span class="st">" "</span>, <span class="st">"^"</span>, REV, <span class="st">"..."</span>, FWD.RC)</span>
<span id="cb7-52"><a href="#cb7-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-53"><a href="#cb7-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Run cutadapt itself</span></span>
<span id="cb7-54"><a href="#cb7-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-55"><a href="#cb7-55" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {</span>
<span id="cb7-56"><a href="#cb7-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(</span>
<span id="cb7-57"><a href="#cb7-57" aria-hidden="true" tabindex="-1"></a>    R1.flags, R2.flags, <span class="st">"-n"</span>, <span class="dv">2</span>, <span class="st">"-m"</span>, <span class="dv">1</span>,</span>
<span id="cb7-58"><a href="#cb7-58" aria-hidden="true" tabindex="-1"></a>    <span class="st">"--discard-untrimmed"</span>,</span>
<span id="cb7-59"><a href="#cb7-59" aria-hidden="true" tabindex="-1"></a>    <span class="st">"-j"</span>, <span class="dv">0</span>,</span>
<span id="cb7-60"><a href="#cb7-60" aria-hidden="true" tabindex="-1"></a>    <span class="st">"-o"</span>, fnFs.cut[i], <span class="st">"-p"</span>, fnRs.cut[i], <span class="co"># output files</span></span>
<span id="cb7-61"><a href="#cb7-61" aria-hidden="true" tabindex="-1"></a>    filtFs[i], filtRs[i], <span class="co"># input files</span></span>
<span id="cb7-62"><a href="#cb7-62" aria-hidden="true" tabindex="-1"></a>    <span class="st">"--report=minimal"</span></span>
<span id="cb7-63"><a href="#cb7-63" aria-hidden="true" tabindex="-1"></a>  )) <span class="co"># Report minimal reports a summary</span></span>
<span id="cb7-64"><a href="#cb7-64" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb7-65"><a href="#cb7-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-66"><a href="#cb7-66" aria-hidden="true" tabindex="-1"></a><span class="co">#  -n 2: remove the primers</span></span>
<span id="cb7-67"><a href="#cb7-67" aria-hidden="true" tabindex="-1"></a><span class="co">#   -m 1: remove empty reads</span></span>
<span id="cb7-68"><a href="#cb7-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-69"><a href="#cb7-69" aria-hidden="true" tabindex="-1"></a><span class="co"># cutadapt will remove the primers and also the sequences in which none of the primers are found</span></span>
<span id="cb7-70"><a href="#cb7-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-71"><a href="#cb7-71" aria-hidden="true" tabindex="-1"></a><span class="co"># after running cutadapt, check the number of primers found now in the output reads, in sample number "1"</span></span>
<span id="cb7-72"><a href="#cb7-72" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(</span>
<span id="cb7-73"><a href="#cb7-73" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">1</span>]]),</span>
<span id="cb7-74"><a href="#cb7-74" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">1</span>]]),</span>
<span id="cb7-75"><a href="#cb7-75" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">1</span>]]),</span>
<span id="cb7-76"><a href="#cb7-76" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">1</span>]])</span>
<span id="cb7-77"><a href="#cb7-77" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb7-78"><a href="#cb7-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-79"><a href="#cb7-79" aria-hidden="true" tabindex="-1"></a><span class="co"># be CAREFUL! it could be possible to find still some primers in your dataset. Do not worry if there are no many. Cutadapt and primerHits function</span></span>
<span id="cb7-80"><a href="#cb7-80" aria-hidden="true" tabindex="-1"></a><span class="co"># do not work in the same way, so some primers can be found.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-6-dada2" class="level1">
<h1>Step 6: DADA2</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># (visit DADA2 website for more information about the specific steps)</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Learn error rates................</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(fnFs.cut, <span class="at">multithread =</span> T, <span class="at">verbose =</span> <span class="dv">1</span>)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>errR <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(fnRs.cut, <span class="at">multithread =</span> T, <span class="at">verbose =</span> <span class="dv">1</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="co"># View Error plots................</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errR, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample inference.................</span></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co"># it calculates the inferred composition of the samples. Removes all the possible sequence errors</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a><span class="co"># and keeps just the real sequences.</span></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">&lt;-</span> <span class="fu">dada</span>(fnFs.cut, <span class="at">err =</span> errF, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>dadaRs <span class="ot">&lt;-</span> <span class="fu">dada</span>(fnRs.cut, <span class="at">err =</span> errR, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>dadaFs[[<span class="dv">5</span>]] <span class="co"># check the inferred sample number 5. We can see the number of ASVs and unique sequences</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>dadaRs[[<span class="dv">5</span>]]</span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaFs) <span class="ot">&lt;-</span> sample.names <span class="co"># giving the correct name of the samples</span></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaRs) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb8-21"><a href="#cb8-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-22"><a href="#cb8-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Merging F and R reads.............................</span></span>
<span id="cb8-23"><a href="#cb8-23" aria-hidden="true" tabindex="-1"></a>mergers <span class="ot">&lt;-</span> <span class="fu">mergePairs</span>(dadaFs, fnFs.cut, dadaRs, fnRs.cut, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb8-24"><a href="#cb8-24" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mergers[[<span class="dv">5</span>]]) <span class="co"># check the results for sample number 5</span></span>
<span id="cb8-25"><a href="#cb8-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-26"><a href="#cb8-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the sequence tables......................</span></span>
<span id="cb8-27"><a href="#cb8-27" aria-hidden="true" tabindex="-1"></a>seqtab_primavera_rizo <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(mergers)</span>
<span id="cb8-28"><a href="#cb8-28" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab_primavera_rizo) <span class="co"># it shows the number of samples (including MOCK samples, negative control and so on) and the number of ASVs</span></span>
<span id="cb8-29"><a href="#cb8-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-30"><a href="#cb8-30" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab_primavera_rizo, <span class="st">"C:/Users/radik/Desktop/Psylvestris_DataPper/seqtab_primavera_rizo_16S.rds"</span>)</span>
<span id="cb8-31"><a href="#cb8-31" aria-hidden="true" tabindex="-1"></a><span class="co"># save the corresponding RDS file</span></span>
<span id="cb8-32"><a href="#cb8-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-33"><a href="#cb8-33" aria-hidden="true" tabindex="-1"></a><span class="co"># IMPORTANT: if we have different sequencing runs included in the same scientific project, we should merge then</span></span>
<span id="cb8-34"><a href="#cb8-34" aria-hidden="true" tabindex="-1"></a><span class="co"># at this point. For that purpose, we should merge the sequence tables coming from each sequencing runs</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-7-merge-seqtabs-obtained-from-different-sequencing-runs" class="level1">
<h1>Step 7: Merge SeqTabs obtained from different sequencing runs</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># (skip this step if you included all the samples in one sequencing run. In that case, go to step7)</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>seqtab_pri_endo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"seqtab_primavera_endo_16s.rds"</span>) <span class="co"># load each sequence table</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>seqtab_pri_rizo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"seqtab_primavera_rizo_16S.rds"</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>seqtab_ve_endo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"seqtab_verano_endo_16s.rds"</span>)</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>seqtab_ve_endo_rizo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"seqtab_verano_endo_rizo_16s.rds"</span>)</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>seqtab_ve_rizo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"seqtab_verano_rizo_16S.rds"</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a><span class="co"># merge all the seqTabs</span></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>mergedSeqTab <span class="ot">&lt;-</span> <span class="fu">mergeSequenceTables</span>(seqtab_pri_endo, seqtab_pri_rizo,</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  seqtab_ve_endo, seqtab_ve_endo_rizo, seqtab_ve_rizo,</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">repeats =</span> <span class="st">"sum"</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-8-chimera-removal" class="level1">
<h1>Step 8: Chimera removal</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(mergedSeqTab, <span class="at">method =</span> <span class="st">"consensus"</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># in case you have just one seqtab, replace "mergedSeqTab" by "seqtab_primavera_rizo"</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim) <span class="co"># indicates the number of samples and ASVs, but not the number of sequences</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the number of ASVs, length</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim))) <span class="co"># Number of ASV of each length</span></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co"># first line: length (bp)</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="co"># second line: number of ASVs of that length</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the number of sequences of each ASV</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen <span class="ot">&lt;-</span> <span class="fu">tapply</span>(<span class="fu">colSums</span>(seqtab.nochim), <span class="fu">factor</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim))), sum)</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>table_reads_seqlen <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">length =</span> <span class="fu">as.numeric</span>(<span class="fu">names</span>(reads.per.seqlen)), <span class="at">count =</span> reads.per.seqlen)</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> table_reads_seqlen, <span class="fu">aes</span>(<span class="at">x =</span> length, <span class="at">y =</span> count)) <span class="sc">+</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>()</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the length of the sequences</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a><span class="co"># (suggestion: select that length range accounting for many sequences)</span></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">&lt;-</span> seqtab.nochim[, <span class="fu">nchar</span>(<span class="fu">colnames</span>(seqtab.nochim)) <span class="sc">%in%</span> <span class="fu">seq</span>(<span class="dv">402</span>, <span class="dv">428</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-9-check-the-number-of-sequences-retained-in-each-step-of-the-processing" class="level1">
<h1>Step 9: Check the number of sequences retained in each step of the processing</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># IMPORTANT: this step can be made just in case we have one seqtab (one sequencing run)</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Otherwise, skip this to the step 10)</span></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>getN <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">getUniques</span>(x))</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>track <span class="ot">&lt;-</span> <span class="fu">cbind</span>(out, <span class="fu">sapply</span>(dadaFs, getN), <span class="fu">sapply</span>(dadaRs, getN), <span class="fu">sapply</span>(mergers, getN), <span class="fu">rowSums</span>(seqtab.nochim))</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(track) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"input"</span>, <span class="st">"filtered"</span>, <span class="st">"denoisedF"</span>, <span class="st">"denoisedR"</span>, <span class="st">"merged"</span>, <span class="st">"nonchim"</span>)</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(track) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="co"># get the percentages</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>track <span class="ot">&lt;-</span> track <span class="sc">%&gt;%</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.filtered =</span> filtered <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> input,</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.denoisedF =</span> denoisedF <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> filtered,</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.denoisedR =</span> denoisedR <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> filtered,</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.merged =</span> merged <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> filtered,</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.nonchim =</span> nonchim <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> merged,</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.retained =</span> nonchim <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> input</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(track) <span class="co"># here we could see the % of the sequences obtained after each step</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(track, <span class="at">file =</span> <span class="st">"Sequencing_quality.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> T, <span class="at">col.names =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-10-taxonomical-classification" class="level1">
<h1>Step 10: Taxonomical classification</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>taxa_rdp <span class="ot">&lt;-</span> <span class="fu">assignTaxonomy</span>(seqtab.nochim, <span class="st">"/mnt/datos/databases/rdp_train_set_18_H.fa"</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co"># indicate the path where the database is located.</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-11-obtaining-the-asv-table" class="level1">
<h1>Step 11: Obtaining the ASV table</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Format the data</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>ASV <span class="ot">&lt;-</span> seqtab.nochim</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>ASVt <span class="ot">&lt;-</span> <span class="fu">t</span>(ASV)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace NA values by "unclassified" and remove the column "species" created during the classification</span></span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>taxa_rdp_na <span class="ot">&lt;-</span> <span class="fu">apply</span>(taxa_rdp, <span class="dv">2</span>, tidyr<span class="sc">::</span>replace_na, <span class="st">"unclassified"</span>)[, <span class="sc">-</span><span class="dv">7</span>]</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename the ASVs (if we have 100 ASVs, we will name them from ASV001 to ASV100)</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>number.digit <span class="ot">&lt;-</span> <span class="fu">nchar</span>(<span class="fu">as.integer</span>(<span class="fu">nrow</span>(ASVt)))</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"ASV%0"</span>, number.digit, <span class="st">"d"</span>) <span class="co"># As many 0 as digits</span></span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>ASV_names <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(names, <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(ASVt))</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Join taxonomy and ASV table</span></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(taxa_rdp_na, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>), <span class="fu">as.data.frame</span>(ASV_names, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>), <span class="fu">as.data.frame</span>(ASVt, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>))</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the sequence of each ASV to the ASV table in the correct position</span></span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>ASV_seqs <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ASV_table_classified_raw)</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ASV_table_classified_raw) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">&lt;-</span> <span class="fu">cbind</span>(ASV_seqs, ASV_table_classified_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-12-mock-community-quality-control" class="level1">
<h1>Step 12: Mock community quality control</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Eliminacion de la MOCK---------------------------------</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>MockCommunity <span class="ot">&lt;-</span> <span class="cf">function</span>(data, MOCK_composition, ASV_column, <span class="at">choose.first =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get total number of sequences for each MOCK ASV</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  sum_ASVs_MOCK <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(data[, <span class="fu">grep</span>(<span class="st">"MOCK"</span>, <span class="fu">colnames</span>(data)), <span class="at">drop =</span> <span class="cn">FALSE</span>])</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Bind it to ASV_table</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  ASV_table_counts_MOCK <span class="ot">&lt;-</span> <span class="fu">cbind</span>(data, sum_ASVs_MOCK)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(ASV_table_counts_MOCK)[<span class="fu">ncol</span>(ASV_table_counts_MOCK)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Total counts MOCK"</span>)</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sort ASV table according to MOCK ASV rel.abundance</span></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>  ASV_table_counts_MOCK_sorted <span class="ot">&lt;-</span> ASV_table_counts_MOCK[<span class="fu">order</span>(ASV_table_counts_MOCK<span class="sc">$</span><span class="st">"Total counts MOCK"</span>, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate percentage</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a>  percentage <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(ASV_table_counts_MOCK_sorted)) {</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">isTRUE</span>(<span class="fu">any</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span>Genus[i] <span class="sc">%in%</span> MOCK_composition[, <span class="dv">1</span>]))) { <span class="co"># for each line, if Genus is equal to any of the MOCK members,continue with the next line (next)</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a>      <span class="cf">next</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">isTRUE</span>(choose.first)) {</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a>        percentage <span class="ot">&lt;-</span> (ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span>)) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a>          <span class="st">"First ASV found that does not belong to the MOCK community! It is"</span>, ASV_table_counts_MOCK_sorted[i, ][[ASV_column]], <span class="st">"which classifies as"</span>, ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span>Genus, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"and represents a"</span>,</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a>          <span class="fu">round</span>((ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span>)) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">6</span>),</span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a>          <span class="st">"perc. of the sequences. This ASV was used to calculate the percentage"</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>      <span class="co"># if it finds a ASV which does not belong to the MOCK community, make a question to user.</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>      answer <span class="ot">&lt;-</span> <span class="fu">readline</span>(<span class="at">prompt =</span> <span class="fu">cat</span>(</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>        ASV_table_counts_MOCK_sorted[i, ][[ASV_column]],</span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>        <span class="st">"does not belong to the MOCK community."</span>,</span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>        <span class="st">"It representes a "</span>,</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>        <span class="fu">round</span>((ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span>)) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">6</span>),</span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>        <span class="st">" perc. of the sequences"</span>, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"and it classifies as"</span>, ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span>Genus, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"Do you want to use this ASV to calculate the percentage?"</span>, <span class="st">"[answer yes or no]"</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>      ))</span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (answer <span class="sc">==</span> <span class="st">"no"</span>) { <span class="co"># if the user chooses not to use the first spurious ASV, go to the next one</span></span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>        <span class="cf">next</span></span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (answer <span class="sc">==</span> <span class="st">"yes"</span>) { <span class="co"># when the user says "yes", store de percentage and print the ASV name, classification and %</span></span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>        percentage <span class="ot">&lt;-</span> (ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span>)) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(</span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>          <span class="st">"You made a decision!"</span>, ASV_table_counts_MOCK_sorted[i, ][[ASV_column]], <span class="st">"which classifies as"</span>, ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span>Genus, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"and represents a"</span>,</span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>          <span class="fu">round</span>((ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span>)) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">6</span>),</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>          <span class="st">"perc. of the sequences, was used to calculate the percentage"</span></span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span></span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">"Error: Answers have to fit  'yes' or 'no'"</span>)</span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove MOCK columns</span></span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a>  ASV_table_without_MOCK <span class="ot">&lt;-</span> ASV_table_counts_MOCK_sorted[, <span class="sc">-</span><span class="fu">grep</span>(<span class="st">"MOCK"</span>, <span class="fu">colnames</span>(ASV_table_counts_MOCK_sorted))]</span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get number of sequences of each ASV without MOCK</span></span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(ASV_table_without_MOCK) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb14-62"><a href="#cb14-62" aria-hidden="true" tabindex="-1"></a>  ASV_sums <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(ASV_table_without_MOCK[, <span class="dv">9</span><span class="sc">:</span><span class="fu">ncol</span>(ASV_table_without_MOCK)])</span>
<span id="cb14-63"><a href="#cb14-63" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get total number of sequences</span></span>
<span id="cb14-64"><a href="#cb14-64" aria-hidden="true" tabindex="-1"></a>  sum.total <span class="ot">&lt;-</span> <span class="fu">sum</span>(ASV_sums)</span>
<span id="cb14-65"><a href="#cb14-65" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply percentage to sequence number</span></span>
<span id="cb14-66"><a href="#cb14-66" aria-hidden="true" tabindex="-1"></a>  nseq_cutoff <span class="ot">&lt;-</span> (percentage <span class="sc">/</span> <span class="dv">100</span>) <span class="sc">*</span> sum.total</span>
<span id="cb14-67"><a href="#cb14-67" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter table.</span></span>
<span id="cb14-68"><a href="#cb14-68" aria-hidden="true" tabindex="-1"></a>  ASV_filtered <span class="ot">&lt;-</span> ASV_table_without_MOCK[<span class="fu">which</span>(ASV_sums <span class="sc">&gt;</span> nseq_cutoff), ]</span>
<span id="cb14-69"><a href="#cb14-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-70"><a href="#cb14-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-71"><a href="#cb14-71" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sort table in ascending order of ASV names</span></span>
<span id="cb14-72"><a href="#cb14-72" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_sorted <span class="ot">&lt;-</span> ASV_filtered[<span class="fu">order</span>(ASV_filtered[[ASV_column]]), ]</span>
<span id="cb14-73"><a href="#cb14-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(ASV_filtered_sorted)</span>
<span id="cb14-74"><a href="#cb14-74" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-75"><a href="#cb14-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-76"><a href="#cb14-76" aria-hidden="true" tabindex="-1"></a>ASV_filtered_MOCK <span class="ot">&lt;-</span> <span class="fu">MockCommunity</span>(ASV_table_classified_raw, mock_composition, <span class="at">ASV_column =</span> <span class="st">"ASV_names"</span>)</span>
<span id="cb14-77"><a href="#cb14-77" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-78"><a href="#cb14-78" aria-hidden="true" tabindex="-1"></a><span class="co"># select the relative abundance of the first ASV that is not included in the Mock Community. That percentage will be considered the detection limit of sequencing</span></span>
<span id="cb14-79"><a href="#cb14-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-80"><a href="#cb14-80" aria-hidden="true" tabindex="-1"></a><span class="co"># IMPORTANT: write down and keep this cut-off value if you will work also with ITS2 or fungal reads, because you will work just with same cut-off</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-13-removal-of-plastids-plants-chloroplast-and-other-artifcats-sequences" class="level1">
<h1>Step 13: Removal of Plastids, Plants, Chloroplast and other artifcats sequences</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>ASV_final <span class="ot">&lt;-</span> ASV_filtered_MOCK[(<span class="fu">which</span>(ASV_filtered_MOCK<span class="sc">$</span>Genus <span class="sc">!=</span> <span class="st">"Streptophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Genus <span class="sc">!=</span> <span class="st">"Chlorophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Genus <span class="sc">!=</span> <span class="st">"Bacillariophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Family <span class="sc">!=</span> <span class="st">"Streptophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Family <span class="sc">!=</span> <span class="st">"Chlorophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Family <span class="sc">!=</span> <span class="st">"Bacillariophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Family <span class="sc">!=</span> <span class="st">"Mitochondria"</span> <span class="sc">&amp;</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Class <span class="sc">!=</span> <span class="st">"Chloroplast"</span> <span class="sc">&amp;</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Order <span class="sc">!=</span> <span class="st">"Chloroplast"</span> <span class="sc">&amp;</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Kingdom <span class="sc">!=</span> <span class="st">"Eukaryota"</span> <span class="sc">&amp;</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Kingdom <span class="sc">!=</span> <span class="st">"unclassified"</span>)), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="step-14-refinement-of-the-taxonomy" class="level1">
<h1>Step 14: Refinement of the taxonomy</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the taxonomy of that ASVs classified at Phylum level as "Cyanobacteria/Chloroplast"</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(ASV_final[<span class="fu">which</span>(ASV_final<span class="sc">$</span>Phylum <span class="sc">==</span> <span class="st">"Cyanobacteria/Chloroplast"</span>), ])</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Suggestion: if they are classified at Class level properly (as a Cyanobacteria), keep them</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>ASV_final2 <span class="ot">&lt;-</span> ASV_final[(<span class="fu">which</span>(ASV_final<span class="sc">$</span>Phylum <span class="sc">!=</span> <span class="st">"Cyanobacteria/Chloroplast"</span> <span class="sc">|</span> ASV_final<span class="sc">$</span>Class <span class="sc">!=</span> <span class="st">"unclassified"</span>)), ]</span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the sequences of those ASV classified as "Cyanobacteria/Chloroplast"</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>cianos <span class="ot">&lt;-</span> ASV_final2[<span class="fu">which</span>(ASV_final2<span class="sc">$</span>Phylum <span class="sc">==</span> <span class="st">"Cyanobacteria/Chloroplast"</span>), ]</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>seq_cianos <span class="ot">&lt;-</span> <span class="fu">as.list</span>(cianos<span class="sc">$</span>ASV_seqs)</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_cianos, <span class="at">names =</span> cianos<span class="sc">$</span>ASV_names, <span class="at">file.out =</span> <span class="st">"cianos.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span> <span class="dv">1000</span>, <span class="at">as.string =</span> <span class="cn">FALSE</span>)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a><span class="co"># align the corresponding sequences by BLAST and compare them with those held in the nt database of NCBI</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>((<span class="st">"/home/programs/ncbi-blast-2.11.0+/bin/blastn -query cianos.fas -db /mnt/datos/databases/nt -out cianos_hits.txt -outfmt '6 std stitle' -show_gis -max_target_seqs 20 -parse_deflines -num_threads 10"</span>), <span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a><span class="co"># (we verified that all the sequences corresponded to real cyanobacteria, so we did not remove them from the dataset)</span></span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Repeat the same with sequences not classified at Phylum level, since they can be artifacts,</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a><span class="co"># even sequences of the plant host (specially if you are working with endosphere or plant tissues)</span></span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>unclassified_phyla <span class="ot">&lt;-</span> ASV_final2[<span class="fu">which</span>(ASV_final2<span class="sc">$</span>Phylum <span class="sc">==</span> <span class="st">"unclassified"</span>), ]</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>seq_unclassified_phyla <span class="ot">&lt;-</span> <span class="fu">as.list</span>(unclassified_phyla<span class="sc">$</span>ASV_seqs)</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_unclassified_phyla, <span class="at">names =</span> unclassified_phyla<span class="sc">$</span>ASV_names, <span class="at">file.out =</span> <span class="st">"unclassified_phyla.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span> <span class="dv">1000</span>, <span class="at">as.string =</span> <span class="cn">FALSE</span>)</span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a><span class="co"># BLAST</span></span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>((<span class="st">"/home/programs/ncbi-blast-2.11.0+/bin/blastn -query unclassified_phyla.fas -db /mnt/datos/databases/nt -out unclassified_phyla_hits.txt -outfmt '6 std stitle' -show_gis -max_target_seqs 20 -parse_deflines -num_threads 10"</span>), <span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a><span class="co"># (we detected that the ASV03686 corresponded to plant host, so we removed it)</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>ASV_final_all_filters <span class="ot">&lt;-</span> ASV_final2[(<span class="fu">which</span>(ASV_final2<span class="sc">$</span>Phylum <span class="sc">!=</span> <span class="st">"Cyanobacteria/Chloroplast"</span>)), ]</span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>ASV_final_all_filters <span class="ot">&lt;-</span> ASV_final_all_filters[(<span class="fu">which</span>(ASV_final_all_filters<span class="sc">$</span>ASV_names <span class="sc">!=</span> <span class="st">"ASV03686"</span>)), ]</span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the definitive ASV table</span></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(ASV_final_all_filters, <span class="at">file =</span> <span class="st">"ASV_final.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


<!-- -->

</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb17" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Sequencing Reads Processing 16S"</span></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> Ana V Lasa &lt;a href="https://orcid.org/0000-0003-3783-7157" target="orcid.widget"&gt; &lt;img alt="ORCID logo" src="https://info.orcid.org/wp-content/uploads/2019/11/orcid_16x16.png" width="16" height="16" /&gt;&lt;/a&gt;</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span><span class="co"> </span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="co">  eval: false</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a><span class="an">editor_options:</span><span class="co"> </span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co">  chunk_output_type: console</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="fu"># Aim {.unnumbered}</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>Processing of Illumina (MiSeq reads). 16S (bacteria) reads</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"dada2"</span>)</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a><span class="co"># phyloseq</span></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"phyloseq"</span>)</span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Biocgenerics</span></span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"BiocGenerics"</span>)</span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Biostrings</span></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"Biostrings"</span>)</span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a><span class="co"># ShortRead</span></span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">"BiocManager"</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">"BiocManager"</span>)</span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"ShortRead"</span>)</span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot2 (for plotting)</span></span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a><span class="co"># tidyverse</span></span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">"tidyverse"</span>)</span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a><span class="co"># Load corresponding libraries</span></span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dada2)</span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(phyloseq)</span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(BiocGenerics)</span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ShortRead)</span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a>path <span class="ot">&lt;-</span> <span class="st">"C:/Users/radik/Desktop/Psylvestris_DataPper"</span> <span class="co">#set the path where the fastq.gz files are</span></span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a><span class="fu">list.files</span>(path)<span class="co">#check that all the needed files are in the path</span></span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a><span class="co">#NOTE: if our scientific project includes different sequencing runs (for instance, if you have a lot of samples that do not fit well into just one run), </span></span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a><span class="co">#we should repeat steps 1) to 6) for each sequencing run, and then (step 7), merge all the sequence tables obtained in the step 6)</span></span>
<span id="cb17-75"><a href="#cb17-75" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-76"><a href="#cb17-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-77"><a href="#cb17-77" aria-hidden="true" tabindex="-1"></a>If our scientific project includes different sequencing runs (for instance, if you have a lot of samples that do not fit well into just one run), we should repeat steps 1) to 6) for each sequencing run, and then (step 7), merge all the sequence tables obtained in the step 6). </span>
<span id="cb17-78"><a href="#cb17-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-79"><a href="#cb17-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-80"><a href="#cb17-80" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 1: Adaption of the file names</span></span>
<span id="cb17-81"><a href="#cb17-81" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Sort Forward (F) and Reverse (R) files separatedly</span>
<span id="cb17-82"><a href="#cb17-82" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>F and R fastq filenames should have the following format: SAMPLENAME_R1_001.fastq.gz;SAMPLENAME_R2_001.fastq.gz</span>
<span id="cb17-83"><a href="#cb17-83" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>In this example, the names of F and R file are the following: NGS006-22-EXP4-ITS2-SAMPLENAME_yyy-R1_001.fastq.gz</span>
<span id="cb17-84"><a href="#cb17-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-87"><a href="#cb17-87" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-88"><a href="#cb17-88" aria-hidden="true" tabindex="-1"></a>fnFs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">"_R1_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb17-89"><a href="#cb17-89" aria-hidden="true" tabindex="-1"></a>fnRs <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">list.files</span>(path, <span class="at">pattern =</span> <span class="st">"_R2_001.fastq.gz"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>))</span>
<span id="cb17-90"><a href="#cb17-90" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-91"><a href="#cb17-91" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract sample names, assuming filenames have format: xxx-SAMPLENAME_yyy_R1_001.fastq.gz</span></span>
<span id="cb17-92"><a href="#cb17-92" aria-hidden="true" tabindex="-1"></a>sample.names_raw <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">strsplit</span>(<span class="fu">basename</span>(fnFs), <span class="st">"_"</span>), <span class="st">`</span><span class="at">[</span><span class="st">`</span>, <span class="dv">1</span>)</span>
<span id="cb17-93"><a href="#cb17-93" aria-hidden="true" tabindex="-1"></a><span class="co"># this function splits the string when it finds the first "_" symbol, and keeps the first ("1") part of the string</span></span>
<span id="cb17-94"><a href="#cb17-94" aria-hidden="true" tabindex="-1"></a><span class="co"># output in this example: NGS006-22-EXP5-16S-SAMPLENAME</span></span>
<span id="cb17-95"><a href="#cb17-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-96"><a href="#cb17-96" aria-hidden="true" tabindex="-1"></a>sample.names <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"NGS006-22-EXP5-16S-"</span>, <span class="st">""</span>, sample.names_raw) <span class="co"># replace the xxx part of the string, by nothing (or by the characters that allow you to get the sample names</span></span>
<span id="cb17-97"><a href="#cb17-97" aria-hidden="true" tabindex="-1"></a><span class="co"># output in this example: SAMPLENAME</span></span>
<span id="cb17-98"><a href="#cb17-98" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-99"><a href="#cb17-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-100"><a href="#cb17-100" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 2: Check the quality of the reads</span></span>
<span id="cb17-101"><a href="#cb17-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-104"><a href="#cb17-104" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-105"><a href="#cb17-105" aria-hidden="true" tabindex="-1"></a><span class="co"># a) Check the number of reads</span></span>
<span id="cb17-106"><a href="#cb17-106" aria-hidden="true" tabindex="-1"></a>raw_reads_count <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-107"><a href="#cb17-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-108"><a href="#cb17-108" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fnFs)) {</span>
<span id="cb17-109"><a href="#cb17-109" aria-hidden="true" tabindex="-1"></a>  raw_reads_count <span class="ot">&lt;-</span> <span class="fu">rbind</span>(raw_reads_count, <span class="fu">length</span>(ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs[i])))</span>
<span id="cb17-110"><a href="#cb17-110" aria-hidden="true" tabindex="-1"></a>} <span class="co"># it counts the number of reads in fnFs</span></span>
<span id="cb17-111"><a href="#cb17-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-112"><a href="#cb17-112" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(raw_reads_count) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb17-113"><a href="#cb17-113" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">"_"</span> <span class="ot">=</span> <span class="fu">rownames</span>(raw_reads_count), raw_reads_count)</span>
<span id="cb17-114"><a href="#cb17-114" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(a) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Sample"</span>, <span class="st">"Number_of_reads"</span>)</span>
<span id="cb17-115"><a href="#cb17-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-116"><a href="#cb17-116" aria-hidden="true" tabindex="-1"></a><span class="co"># Check which samples have the highlest and lowest amount of reads</span></span>
<span id="cb17-117"><a href="#cb17-117" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.min</span>(raw_reads_count)], <span class="fu">min</span>(raw_reads_count))</span>
<span id="cb17-118"><a href="#cb17-118" aria-hidden="true" tabindex="-1"></a><span class="fu">cbind</span>(<span class="fu">row.names</span>(raw_reads_count)[<span class="fu">which.max</span>(raw_reads_count)], <span class="fu">max</span>(raw_reads_count))</span>
<span id="cb17-119"><a href="#cb17-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-120"><a href="#cb17-120" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(<span class="fu">data.frame</span>(<span class="st">"_"</span> <span class="ot">=</span> <span class="fu">rownames</span>(raw_reads_count), raw_reads_count), <span class="at">file =</span> <span class="st">"NumberRawReads.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> F)</span>
<span id="cb17-121"><a href="#cb17-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-122"><a href="#cb17-122" aria-hidden="true" tabindex="-1"></a><span class="co"># b) Check the length of the reads</span></span>
<span id="cb17-123"><a href="#cb17-123" aria-hidden="true" tabindex="-1"></a>reads <span class="ot">&lt;-</span> ShortRead<span class="sc">::</span><span class="fu">readFastq</span>(fnFs) <span class="co"># it saves the reads into a new variable</span></span>
<span id="cb17-124"><a href="#cb17-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-125"><a href="#cb17-125" aria-hidden="true" tabindex="-1"></a>uniques <span class="ot">&lt;-</span> <span class="fu">unique</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width) <span class="co"># to get the length of the reads (bp)</span></span>
<span id="cb17-126"><a href="#cb17-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-127"><a href="#cb17-127" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="cn">NULL</span> <span class="co"># it counts the number of reads of each length</span></span>
<span id="cb17-128"><a href="#cb17-128" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(uniques)) {</span>
<span id="cb17-129"><a href="#cb17-129" aria-hidden="true" tabindex="-1"></a>  counts <span class="ot">&lt;-</span> <span class="fu">rbind</span>(counts, <span class="fu">length</span>(<span class="fu">which</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width <span class="sc">==</span> uniques[i])))</span>
<span id="cb17-130"><a href="#cb17-130" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-131"><a href="#cb17-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-132"><a href="#cb17-132" aria-hidden="true" tabindex="-1"></a>histogram <span class="ot">&lt;-</span> <span class="fu">cbind</span>(uniques, counts)</span>
<span id="cb17-133"><a href="#cb17-133" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(histogram) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Seq.length"</span>, <span class="st">"counts"</span>)</span>
<span id="cb17-134"><a href="#cb17-134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-135"><a href="#cb17-135" aria-hidden="true" tabindex="-1"></a><span class="co"># check the histogram</span></span>
<span id="cb17-136"><a href="#cb17-136" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(histogram[<span class="fu">order</span>(histogram[, <span class="dv">1</span>], <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]) <span class="co"># Most sequences should fall in the expected sequence length</span></span>
<span id="cb17-137"><a href="#cb17-137" aria-hidden="true" tabindex="-1"></a><span class="co"># since we followed a 2x275 PE strategy, most of our reads are of 275-276bp</span></span>
<span id="cb17-138"><a href="#cb17-138" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(histogram, <span class="at">file =</span> <span class="st">"Length_Raw_Reads.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> F)</span>
<span id="cb17-139"><a href="#cb17-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-140"><a href="#cb17-140" aria-hidden="true" tabindex="-1"></a><span class="co"># plot the histogram</span></span>
<span id="cb17-141"><a href="#cb17-141" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(reads<span class="sc">@</span>quality<span class="sc">@</span>quality<span class="sc">@</span>ranges<span class="sc">@</span>width, <span class="at">main =</span> <span class="st">"Forward length distribution"</span>, <span class="at">xlab =</span> <span class="st">"Sequence length"</span>, <span class="at">ylab =</span> <span class="st">"Raw reads"</span>)</span>
<span id="cb17-142"><a href="#cb17-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-143"><a href="#cb17-143" aria-hidden="true" tabindex="-1"></a><span class="co"># c) Check the quality plots</span></span>
<span id="cb17-144"><a href="#cb17-144" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnFs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>]) <span class="co"># here you can plot whichever of the samples you want (in this example, samples number 4 and 5)</span></span>
<span id="cb17-145"><a href="#cb17-145" aria-hidden="true" tabindex="-1"></a><span class="fu">plotQualityProfile</span>(fnRs[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb17-146"><a href="#cb17-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-147"><a href="#cb17-147" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-148"><a href="#cb17-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-149"><a href="#cb17-149" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 3. FIGARO</span></span>
<span id="cb17-150"><a href="#cb17-150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-153"><a href="#cb17-153" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-154"><a href="#cb17-154" aria-hidden="true" tabindex="-1"></a><span class="co"># Preparation of the data</span></span>
<span id="cb17-155"><a href="#cb17-155" aria-hidden="true" tabindex="-1"></a><span class="co"># Creamos una carpeta donde guarde todas las secuencias tras el corte por figaro</span></span>
<span id="cb17-156"><a href="#cb17-156" aria-hidden="true" tabindex="-1"></a>figFs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"figaro"</span>, <span class="fu">basename</span>(fnFs))</span>
<span id="cb17-157"><a href="#cb17-157" aria-hidden="true" tabindex="-1"></a>figRs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"figaro"</span>, <span class="fu">basename</span>(fnRs))</span>
<span id="cb17-158"><a href="#cb17-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-159"><a href="#cb17-159" aria-hidden="true" tabindex="-1"></a><span class="co"># Figaro requires all the reads are of the same length, so firstly we have to cut the sequences trying to</span></span>
<span id="cb17-160"><a href="#cb17-160" aria-hidden="true" tabindex="-1"></a><span class="co"># include all the sequences, so we make an initial cut at 274 bp (we won't lose many reads)</span></span>
<span id="cb17-161"><a href="#cb17-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-162"><a href="#cb17-162" aria-hidden="true" tabindex="-1"></a>out.figaro <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(fnFs, figFs, fnRs, figRs, <span class="at">compress =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>, <span class="at">truncLen =</span> <span class="fu">c</span>(<span class="dv">274</span>, <span class="dv">274</span>))</span>
<span id="cb17-163"><a href="#cb17-163" aria-hidden="true" tabindex="-1"></a><span class="co"># filterAndTrim(input F, output F, input R, output R)</span></span>
<span id="cb17-164"><a href="#cb17-164" aria-hidden="true" tabindex="-1"></a><span class="co"># IMPORTANT: figaro will consider these reads just for calculation of the best parameters for DADA2</span></span>
<span id="cb17-165"><a href="#cb17-165" aria-hidden="true" tabindex="-1"></a><span class="co"># but these sequences will not be used for DADA2, we then will use all the reads for DADA2</span></span>
<span id="cb17-166"><a href="#cb17-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-167"><a href="#cb17-167" aria-hidden="true" tabindex="-1"></a>FWD <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"CCTACGGGNBGCASCAG"</span>) <span class="co"># insert here the sequence of the the primer F</span></span>
<span id="cb17-168"><a href="#cb17-168" aria-hidden="true" tabindex="-1"></a>REV <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"GACTACNVGGGTATCTAATCC"</span>) <span class="co"># here the sequence of the primer R</span></span>
<span id="cb17-169"><a href="#cb17-169" aria-hidden="true" tabindex="-1"></a><span class="fu">nchar</span>(FWD) <span class="co"># to determine the number of bp of the primers F and R</span></span>
<span id="cb17-170"><a href="#cb17-170" aria-hidden="true" tabindex="-1"></a><span class="fu">nchar</span>(REV)</span>
<span id="cb17-171"><a href="#cb17-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-172"><a href="#cb17-172" aria-hidden="true" tabindex="-1"></a><span class="co"># Run figaro definitely:</span></span>
<span id="cb17-173"><a href="#cb17-173" aria-hidden="true" tabindex="-1"></a>figaro <span class="ot">&lt;-</span> <span class="fu">system</span>((<span class="st">"python3 /home/programs/figaro/figaro/figaro.py -i C:/Users/radik/Desktop/Psylvestris_DataPper/figaro -o C:/Users/radik/Desktop/Psylvestris_DataPper/figaro -a 426 -f 17 -r 21"</span>),</span>
<span id="cb17-174"><a href="#cb17-174" aria-hidden="true" tabindex="-1"></a>  <span class="at">intern =</span> <span class="cn">TRUE</span></span>
<span id="cb17-175"><a href="#cb17-175" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-176"><a href="#cb17-176" aria-hidden="true" tabindex="-1"></a><span class="co">#-i insert the path of the previously created figaro folder</span></span>
<span id="cb17-177"><a href="#cb17-177" aria-hidden="true" tabindex="-1"></a><span class="co">#-o the path of the figaro folder</span></span>
<span id="cb17-178"><a href="#cb17-178" aria-hidden="true" tabindex="-1"></a><span class="co">#-a amplicon size (bp) according to the 16S rRNA hypervariable regions under study</span></span>
<span id="cb17-179"><a href="#cb17-179" aria-hidden="true" tabindex="-1"></a><span class="co">#-f size of the F primer (bp)</span></span>
<span id="cb17-180"><a href="#cb17-180" aria-hidden="true" tabindex="-1"></a><span class="co">#-r size of the R primer (bp)</span></span>
<span id="cb17-181"><a href="#cb17-181" aria-hidden="true" tabindex="-1"></a><span class="co"># figaro script (figaro.py) runs properly in python3</span></span>
<span id="cb17-182"><a href="#cb17-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-183"><a href="#cb17-183" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(figaro) <span class="co"># check the results of figaro and select that maxExpectedError with which you obtain a good score and a good percentage of retained reads</span></span>
<span id="cb17-184"><a href="#cb17-184" aria-hidden="true" tabindex="-1"></a><span class="co"># write down the maxEE and the trimming positions, in this example: trimming positions [F,R]: [269,215]; maxEE[F,R]: [2,2]</span></span>
<span id="cb17-185"><a href="#cb17-185" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-186"><a href="#cb17-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-187"><a href="#cb17-187" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 4. Filter and trimming steps </span></span>
<span id="cb17-188"><a href="#cb17-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-191"><a href="#cb17-191" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-192"><a href="#cb17-192" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the folder "filtered" to save the trimmed reads</span></span>
<span id="cb17-193"><a href="#cb17-193" aria-hidden="true" tabindex="-1"></a>filtFs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnFs))</span>
<span id="cb17-194"><a href="#cb17-194" aria-hidden="true" tabindex="-1"></a>filtRs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"filtered"</span>, <span class="fu">basename</span>(fnRs))</span>
<span id="cb17-195"><a href="#cb17-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-196"><a href="#cb17-196" aria-hidden="true" tabindex="-1"></a><span class="co"># trimming step itself:</span></span>
<span id="cb17-197"><a href="#cb17-197" aria-hidden="true" tabindex="-1"></a>out <span class="ot">&lt;-</span> <span class="fu">filterAndTrim</span>(fnFs, filtFs, fnRs, filtRs,</span>
<span id="cb17-198"><a href="#cb17-198" aria-hidden="true" tabindex="-1"></a>  <span class="at">truncLen =</span> <span class="fu">c</span>(<span class="dv">269</span>, <span class="dv">215</span>),</span>
<span id="cb17-199"><a href="#cb17-199" aria-hidden="true" tabindex="-1"></a>  <span class="at">maxN =</span> <span class="dv">0</span>, <span class="at">maxEE =</span> <span class="fu">c</span>(<span class="dv">2</span>, <span class="dv">2</span>), <span class="at">truncQ =</span> <span class="dv">2</span>, <span class="at">rm.phix =</span> <span class="cn">TRUE</span>,</span>
<span id="cb17-200"><a href="#cb17-200" aria-hidden="true" tabindex="-1"></a>  <span class="at">compress =</span> <span class="cn">TRUE</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>, <span class="at">minLen =</span> <span class="dv">50</span></span>
<span id="cb17-201"><a href="#cb17-201" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-202"><a href="#cb17-202" aria-hidden="true" tabindex="-1"></a><span class="co"># input reads F, output reads F, input reads R, output reads R,</span></span>
<span id="cb17-203"><a href="#cb17-203" aria-hidden="true" tabindex="-1"></a><span class="co"># truncLen: trimming positions in F and R reads, according to Figaro results</span></span>
<span id="cb17-204"><a href="#cb17-204" aria-hidden="true" tabindex="-1"></a><span class="co"># maxN: number of allowed ambiguities (N) in the reads</span></span>
<span id="cb17-205"><a href="#cb17-205" aria-hidden="true" tabindex="-1"></a><span class="co"># maxEE: maximum expected errors in F and R reads, according to Figaro results</span></span>
<span id="cb17-206"><a href="#cb17-206" aria-hidden="true" tabindex="-1"></a><span class="co"># trunqQ: the minimum quality score of each nucleotides; remove reads with at least one nucleotide with an associated quality score under the selected value</span></span>
<span id="cb17-207"><a href="#cb17-207" aria-hidden="true" tabindex="-1"></a><span class="co"># minLen: the minimum size of the reads; remove all the reads under this size (bp)</span></span>
<span id="cb17-208"><a href="#cb17-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-209"><a href="#cb17-209" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(out)</span>
<span id="cb17-210"><a href="#cb17-210" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-211"><a href="#cb17-211" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-212"><a href="#cb17-212" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 5: CUTADAPT. primer removal from F and R reads</span></span>
<span id="cb17-213"><a href="#cb17-213" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-216"><a href="#cb17-216" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-217"><a href="#cb17-217" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a function to calculate all the orientations of F and R primers</span></span>
<span id="cb17-218"><a href="#cb17-218" aria-hidden="true" tabindex="-1"></a>allOrients <span class="ot">&lt;-</span> <span class="cf">function</span>(primer) {</span>
<span id="cb17-219"><a href="#cb17-219" aria-hidden="true" tabindex="-1"></a>  <span class="fu">require</span>(Biostrings)</span>
<span id="cb17-220"><a href="#cb17-220" aria-hidden="true" tabindex="-1"></a>  dna <span class="ot">&lt;-</span> <span class="fu">DNAString</span>(primer) <span class="co"># BioStrings package works with strings but not with characters</span></span>
<span id="cb17-221"><a href="#cb17-221" aria-hidden="true" tabindex="-1"></a>  orients <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb17-222"><a href="#cb17-222" aria-hidden="true" tabindex="-1"></a>    <span class="at">Forward =</span> dna, <span class="at">Complement =</span> Biostrings<span class="sc">::</span><span class="fu">complement</span>(dna), <span class="at">Reverse =</span> <span class="fu">reverse</span>(dna),</span>
<span id="cb17-223"><a href="#cb17-223" aria-hidden="true" tabindex="-1"></a>    <span class="at">RevComp =</span> <span class="fu">reverseComplement</span>(dna)</span>
<span id="cb17-224"><a href="#cb17-224" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb17-225"><a href="#cb17-225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sapply</span>(orients, toString)) <span class="co"># change to string</span></span>
<span id="cb17-226"><a href="#cb17-226" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-227"><a href="#cb17-227" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(FWD) <span class="co"># pass the function to primer F</span></span>
<span id="cb17-228"><a href="#cb17-228" aria-hidden="true" tabindex="-1"></a>REV.orients <span class="ot">&lt;-</span> <span class="fu">allOrients</span>(REV) <span class="co"># pass the function to primer R</span></span>
<span id="cb17-229"><a href="#cb17-229" aria-hidden="true" tabindex="-1"></a>FWD.orients <span class="co"># check the calculated orientations</span></span>
<span id="cb17-230"><a href="#cb17-230" aria-hidden="true" tabindex="-1"></a>REV.orients</span>
<span id="cb17-231"><a href="#cb17-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-232"><a href="#cb17-232" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of times both primers (in all the possible orientations) appear in our dataset</span></span>
<span id="cb17-233"><a href="#cb17-233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-234"><a href="#cb17-234" aria-hidden="true" tabindex="-1"></a>primerHits <span class="ot">&lt;-</span> <span class="cf">function</span>(primer, fn) {</span>
<span id="cb17-235"><a href="#cb17-235" aria-hidden="true" tabindex="-1"></a>  nhits <span class="ot">&lt;-</span> <span class="fu">vcountPattern</span>(primer, <span class="fu">sread</span>(<span class="fu">readFastq</span>(fn)), <span class="at">fixed =</span> <span class="cn">FALSE</span>)</span>
<span id="cb17-236"><a href="#cb17-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">sum</span>(nhits <span class="sc">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb17-237"><a href="#cb17-237" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-238"><a href="#cb17-238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-239"><a href="#cb17-239" aria-hidden="true" tabindex="-1"></a><span class="co"># check the number of reads in which the primers F and R (in all possible orientations) are found in sample number "1"</span></span>
<span id="cb17-240"><a href="#cb17-240" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(</span>
<span id="cb17-241"><a href="#cb17-241" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">1</span>]]),</span>
<span id="cb17-242"><a href="#cb17-242" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">1</span>]]),</span>
<span id="cb17-243"><a href="#cb17-243" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtFs[[<span class="dv">1</span>]]),</span>
<span id="cb17-244"><a href="#cb17-244" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> filtRs[[<span class="dv">1</span>]])</span>
<span id="cb17-245"><a href="#cb17-245" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-246"><a href="#cb17-246" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-247"><a href="#cb17-247" aria-hidden="true" tabindex="-1"></a><span class="co"># On the left, we have the presence of forward and reverse primers (FWD and REV) in forward and reverse reads (FWD.ForwardReads, FWD.ReverseReads, REV.ForwardReads and REV.ReverseReads). On the top of the table, we see the four possible orientations of the primers (Forward, Complement, Reverse and RevComp).</span></span>
<span id="cb17-248"><a href="#cb17-248" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-249"><a href="#cb17-249" aria-hidden="true" tabindex="-1"></a>cutadapt <span class="ot">&lt;-</span> <span class="st">"/usr/local/bin/cutadapt"</span> <span class="co"># path to cutadapt in your computer/machine</span></span>
<span id="cb17-250"><a href="#cb17-250" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-251"><a href="#cb17-251" aria-hidden="true" tabindex="-1"></a><span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(<span class="st">"--version"</span>)) <span class="co"># Run shell commands from R</span></span>
<span id="cb17-252"><a href="#cb17-252" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-253"><a href="#cb17-253" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a directoy or folder where your reads will be saved</span></span>
<span id="cb17-254"><a href="#cb17-254" aria-hidden="true" tabindex="-1"></a>path.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path, <span class="st">"cutadapt"</span>)</span>
<span id="cb17-255"><a href="#cb17-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-256"><a href="#cb17-256" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(path.cut)) <span class="fu">dir.create</span>(path.cut)</span>
<span id="cb17-257"><a href="#cb17-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-258"><a href="#cb17-258" aria-hidden="true" tabindex="-1"></a>fnFs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnFs))</span>
<span id="cb17-259"><a href="#cb17-259" aria-hidden="true" tabindex="-1"></a>fnRs.cut <span class="ot">&lt;-</span> <span class="fu">file.path</span>(path.cut, <span class="fu">basename</span>(fnRs))</span>
<span id="cb17-260"><a href="#cb17-260" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-261"><a href="#cb17-261" aria-hidden="true" tabindex="-1"></a><span class="co"># Produce arguments for cutadapt (visit the cutadapt website for more information)</span></span>
<span id="cb17-262"><a href="#cb17-262" aria-hidden="true" tabindex="-1"></a>FWD.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(FWD) <span class="co"># calculate the reverse complementary sequence</span></span>
<span id="cb17-263"><a href="#cb17-263" aria-hidden="true" tabindex="-1"></a>REV.RC <span class="ot">&lt;-</span> dada2<span class="sc">:::</span><span class="fu">rc</span>(REV)</span>
<span id="cb17-264"><a href="#cb17-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-265"><a href="#cb17-265" aria-hidden="true" tabindex="-1"></a><span class="co"># add the specific adapters to F and R reads, which are need for cutadapt</span></span>
<span id="cb17-266"><a href="#cb17-266" aria-hidden="true" tabindex="-1"></a>R1.flags <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"-a"</span>, <span class="st">" "</span>, <span class="st">"^"</span>, FWD, <span class="st">"..."</span>, REV.RC)</span>
<span id="cb17-267"><a href="#cb17-267" aria-hidden="true" tabindex="-1"></a>R2.flags <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"-A"</span>, <span class="st">" "</span>, <span class="st">"^"</span>, REV, <span class="st">"..."</span>, FWD.RC)</span>
<span id="cb17-268"><a href="#cb17-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-269"><a href="#cb17-269" aria-hidden="true" tabindex="-1"></a><span class="co"># Run cutadapt itself</span></span>
<span id="cb17-270"><a href="#cb17-270" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-271"><a href="#cb17-271" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">seq_along</span>(fnFs)) {</span>
<span id="cb17-272"><a href="#cb17-272" aria-hidden="true" tabindex="-1"></a>  <span class="fu">system2</span>(cutadapt, <span class="at">args =</span> <span class="fu">c</span>(</span>
<span id="cb17-273"><a href="#cb17-273" aria-hidden="true" tabindex="-1"></a>    R1.flags, R2.flags, <span class="st">"-n"</span>, <span class="dv">2</span>, <span class="st">"-m"</span>, <span class="dv">1</span>,</span>
<span id="cb17-274"><a href="#cb17-274" aria-hidden="true" tabindex="-1"></a>    <span class="st">"--discard-untrimmed"</span>,</span>
<span id="cb17-275"><a href="#cb17-275" aria-hidden="true" tabindex="-1"></a>    <span class="st">"-j"</span>, <span class="dv">0</span>,</span>
<span id="cb17-276"><a href="#cb17-276" aria-hidden="true" tabindex="-1"></a>    <span class="st">"-o"</span>, fnFs.cut[i], <span class="st">"-p"</span>, fnRs.cut[i], <span class="co"># output files</span></span>
<span id="cb17-277"><a href="#cb17-277" aria-hidden="true" tabindex="-1"></a>    filtFs[i], filtRs[i], <span class="co"># input files</span></span>
<span id="cb17-278"><a href="#cb17-278" aria-hidden="true" tabindex="-1"></a>    <span class="st">"--report=minimal"</span></span>
<span id="cb17-279"><a href="#cb17-279" aria-hidden="true" tabindex="-1"></a>  )) <span class="co"># Report minimal reports a summary</span></span>
<span id="cb17-280"><a href="#cb17-280" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-281"><a href="#cb17-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-282"><a href="#cb17-282" aria-hidden="true" tabindex="-1"></a><span class="co">#  -n 2: remove the primers</span></span>
<span id="cb17-283"><a href="#cb17-283" aria-hidden="true" tabindex="-1"></a><span class="co">#   -m 1: remove empty reads</span></span>
<span id="cb17-284"><a href="#cb17-284" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-285"><a href="#cb17-285" aria-hidden="true" tabindex="-1"></a><span class="co"># cutadapt will remove the primers and also the sequences in which none of the primers are found</span></span>
<span id="cb17-286"><a href="#cb17-286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-287"><a href="#cb17-287" aria-hidden="true" tabindex="-1"></a><span class="co"># after running cutadapt, check the number of primers found now in the output reads, in sample number "1"</span></span>
<span id="cb17-288"><a href="#cb17-288" aria-hidden="true" tabindex="-1"></a><span class="fu">rbind</span>(</span>
<span id="cb17-289"><a href="#cb17-289" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ForwardReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">1</span>]]),</span>
<span id="cb17-290"><a href="#cb17-290" aria-hidden="true" tabindex="-1"></a>  <span class="at">FWD.ReverseReads =</span> <span class="fu">sapply</span>(FWD.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">1</span>]]),</span>
<span id="cb17-291"><a href="#cb17-291" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ForwardReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnFs.cut[[<span class="dv">1</span>]]),</span>
<span id="cb17-292"><a href="#cb17-292" aria-hidden="true" tabindex="-1"></a>  <span class="at">REV.ReverseReads =</span> <span class="fu">sapply</span>(REV.orients, primerHits, <span class="at">fn =</span> fnRs.cut[[<span class="dv">1</span>]])</span>
<span id="cb17-293"><a href="#cb17-293" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-294"><a href="#cb17-294" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-295"><a href="#cb17-295" aria-hidden="true" tabindex="-1"></a><span class="co"># be CAREFUL! it could be possible to find still some primers in your dataset. Do not worry if there are no many. Cutadapt and primerHits function</span></span>
<span id="cb17-296"><a href="#cb17-296" aria-hidden="true" tabindex="-1"></a><span class="co"># do not work in the same way, so some primers can be found.</span></span>
<span id="cb17-297"><a href="#cb17-297" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-298"><a href="#cb17-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-299"><a href="#cb17-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-300"><a href="#cb17-300" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 6: DADA2</span></span>
<span id="cb17-301"><a href="#cb17-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-304"><a href="#cb17-304" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-305"><a href="#cb17-305" aria-hidden="true" tabindex="-1"></a><span class="co"># (visit DADA2 website for more information about the specific steps)</span></span>
<span id="cb17-306"><a href="#cb17-306" aria-hidden="true" tabindex="-1"></a><span class="co"># Learn error rates................</span></span>
<span id="cb17-307"><a href="#cb17-307" aria-hidden="true" tabindex="-1"></a>errF <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(fnFs.cut, <span class="at">multithread =</span> T, <span class="at">verbose =</span> <span class="dv">1</span>)</span>
<span id="cb17-308"><a href="#cb17-308" aria-hidden="true" tabindex="-1"></a>errR <span class="ot">&lt;-</span> <span class="fu">learnErrors</span>(fnRs.cut, <span class="at">multithread =</span> T, <span class="at">verbose =</span> <span class="dv">1</span>)</span>
<span id="cb17-309"><a href="#cb17-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-310"><a href="#cb17-310" aria-hidden="true" tabindex="-1"></a><span class="co"># View Error plots................</span></span>
<span id="cb17-311"><a href="#cb17-311" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errF, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-312"><a href="#cb17-312" aria-hidden="true" tabindex="-1"></a><span class="fu">plotErrors</span>(errR, <span class="at">nominalQ =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-313"><a href="#cb17-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-314"><a href="#cb17-314" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample inference.................</span></span>
<span id="cb17-315"><a href="#cb17-315" aria-hidden="true" tabindex="-1"></a><span class="co"># it calculates the inferred composition of the samples. Removes all the possible sequence errors</span></span>
<span id="cb17-316"><a href="#cb17-316" aria-hidden="true" tabindex="-1"></a><span class="co"># and keeps just the real sequences.</span></span>
<span id="cb17-317"><a href="#cb17-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-318"><a href="#cb17-318" aria-hidden="true" tabindex="-1"></a>dadaFs <span class="ot">&lt;-</span> <span class="fu">dada</span>(fnFs.cut, <span class="at">err =</span> errF, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-319"><a href="#cb17-319" aria-hidden="true" tabindex="-1"></a>dadaRs <span class="ot">&lt;-</span> <span class="fu">dada</span>(fnRs.cut, <span class="at">err =</span> errR, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-320"><a href="#cb17-320" aria-hidden="true" tabindex="-1"></a>dadaFs[[<span class="dv">5</span>]] <span class="co"># check the inferred sample number 5. We can see the number of ASVs and unique sequences</span></span>
<span id="cb17-321"><a href="#cb17-321" aria-hidden="true" tabindex="-1"></a>dadaRs[[<span class="dv">5</span>]]</span>
<span id="cb17-322"><a href="#cb17-322" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-323"><a href="#cb17-323" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaFs) <span class="ot">&lt;-</span> sample.names <span class="co"># giving the correct name of the samples</span></span>
<span id="cb17-324"><a href="#cb17-324" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dadaRs) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb17-325"><a href="#cb17-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-326"><a href="#cb17-326" aria-hidden="true" tabindex="-1"></a><span class="co"># Merging F and R reads.............................</span></span>
<span id="cb17-327"><a href="#cb17-327" aria-hidden="true" tabindex="-1"></a>mergers <span class="ot">&lt;-</span> <span class="fu">mergePairs</span>(dadaFs, fnFs.cut, dadaRs, fnRs.cut, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-328"><a href="#cb17-328" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mergers[[<span class="dv">5</span>]]) <span class="co"># check the results for sample number 5</span></span>
<span id="cb17-329"><a href="#cb17-329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-330"><a href="#cb17-330" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the sequence tables......................</span></span>
<span id="cb17-331"><a href="#cb17-331" aria-hidden="true" tabindex="-1"></a>seqtab_primavera_rizo <span class="ot">&lt;-</span> <span class="fu">makeSequenceTable</span>(mergers)</span>
<span id="cb17-332"><a href="#cb17-332" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab_primavera_rizo) <span class="co"># it shows the number of samples (including MOCK samples, negative control and so on) and the number of ASVs</span></span>
<span id="cb17-333"><a href="#cb17-333" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-334"><a href="#cb17-334" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(seqtab_primavera_rizo, <span class="st">"C:/Users/radik/Desktop/Psylvestris_DataPper/seqtab_primavera_rizo_16S.rds"</span>)</span>
<span id="cb17-335"><a href="#cb17-335" aria-hidden="true" tabindex="-1"></a><span class="co"># save the corresponding RDS file</span></span>
<span id="cb17-336"><a href="#cb17-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-337"><a href="#cb17-337" aria-hidden="true" tabindex="-1"></a><span class="co"># IMPORTANT: if we have different sequencing runs included in the same scientific project, we should merge then</span></span>
<span id="cb17-338"><a href="#cb17-338" aria-hidden="true" tabindex="-1"></a><span class="co"># at this point. For that purpose, we should merge the sequence tables coming from each sequencing runs</span></span>
<span id="cb17-339"><a href="#cb17-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-340"><a href="#cb17-340" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-341"><a href="#cb17-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-342"><a href="#cb17-342" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 7: Merge SeqTabs obtained from different sequencing runs</span></span>
<span id="cb17-343"><a href="#cb17-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-346"><a href="#cb17-346" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-347"><a href="#cb17-347" aria-hidden="true" tabindex="-1"></a><span class="co"># (skip this step if you included all the samples in one sequencing run. In that case, go to step7)</span></span>
<span id="cb17-348"><a href="#cb17-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-349"><a href="#cb17-349" aria-hidden="true" tabindex="-1"></a>seqtab_pri_endo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"seqtab_primavera_endo_16s.rds"</span>) <span class="co"># load each sequence table</span></span>
<span id="cb17-350"><a href="#cb17-350" aria-hidden="true" tabindex="-1"></a>seqtab_pri_rizo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"seqtab_primavera_rizo_16S.rds"</span>)</span>
<span id="cb17-351"><a href="#cb17-351" aria-hidden="true" tabindex="-1"></a>seqtab_ve_endo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"seqtab_verano_endo_16s.rds"</span>)</span>
<span id="cb17-352"><a href="#cb17-352" aria-hidden="true" tabindex="-1"></a>seqtab_ve_endo_rizo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"seqtab_verano_endo_rizo_16s.rds"</span>)</span>
<span id="cb17-353"><a href="#cb17-353" aria-hidden="true" tabindex="-1"></a>seqtab_ve_rizo <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"seqtab_verano_rizo_16S.rds"</span>)</span>
<span id="cb17-354"><a href="#cb17-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-355"><a href="#cb17-355" aria-hidden="true" tabindex="-1"></a><span class="co"># merge all the seqTabs</span></span>
<span id="cb17-356"><a href="#cb17-356" aria-hidden="true" tabindex="-1"></a>mergedSeqTab <span class="ot">&lt;-</span> <span class="fu">mergeSequenceTables</span>(seqtab_pri_endo, seqtab_pri_rizo,</span>
<span id="cb17-357"><a href="#cb17-357" aria-hidden="true" tabindex="-1"></a>  seqtab_ve_endo, seqtab_ve_endo_rizo, seqtab_ve_rizo,</span>
<span id="cb17-358"><a href="#cb17-358" aria-hidden="true" tabindex="-1"></a>  <span class="at">repeats =</span> <span class="st">"sum"</span></span>
<span id="cb17-359"><a href="#cb17-359" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb17-360"><a href="#cb17-360" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-361"><a href="#cb17-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-362"><a href="#cb17-362" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 8: Chimera removal </span></span>
<span id="cb17-363"><a href="#cb17-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-366"><a href="#cb17-366" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-367"><a href="#cb17-367" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">&lt;-</span> <span class="fu">removeBimeraDenovo</span>(mergedSeqTab, <span class="at">method =</span> <span class="st">"consensus"</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>, <span class="at">verbose =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-368"><a href="#cb17-368" aria-hidden="true" tabindex="-1"></a><span class="co"># in case you have just one seqtab, replace "mergedSeqTab" by "seqtab_primavera_rizo"</span></span>
<span id="cb17-369"><a href="#cb17-369" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(seqtab.nochim) <span class="co"># indicates the number of samples and ASVs, but not the number of sequences</span></span>
<span id="cb17-370"><a href="#cb17-370" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-371"><a href="#cb17-371" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the number of ASVs, length</span></span>
<span id="cb17-372"><a href="#cb17-372" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim))) <span class="co"># Number of ASV of each length</span></span>
<span id="cb17-373"><a href="#cb17-373" aria-hidden="true" tabindex="-1"></a><span class="co"># first line: length (bp)</span></span>
<span id="cb17-374"><a href="#cb17-374" aria-hidden="true" tabindex="-1"></a><span class="co"># second line: number of ASVs of that length</span></span>
<span id="cb17-375"><a href="#cb17-375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-376"><a href="#cb17-376" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the number of sequences of each ASV</span></span>
<span id="cb17-377"><a href="#cb17-377" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen <span class="ot">&lt;-</span> <span class="fu">tapply</span>(<span class="fu">colSums</span>(seqtab.nochim), <span class="fu">factor</span>(<span class="fu">nchar</span>(<span class="fu">getSequences</span>(seqtab.nochim))), sum)</span>
<span id="cb17-378"><a href="#cb17-378" aria-hidden="true" tabindex="-1"></a>reads.per.seqlen</span>
<span id="cb17-379"><a href="#cb17-379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-380"><a href="#cb17-380" aria-hidden="true" tabindex="-1"></a>table_reads_seqlen <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">length =</span> <span class="fu">as.numeric</span>(<span class="fu">names</span>(reads.per.seqlen)), <span class="at">count =</span> reads.per.seqlen)</span>
<span id="cb17-381"><a href="#cb17-381" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> table_reads_seqlen, <span class="fu">aes</span>(<span class="at">x =</span> length, <span class="at">y =</span> count)) <span class="sc">+</span></span>
<span id="cb17-382"><a href="#cb17-382" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>()</span>
<span id="cb17-383"><a href="#cb17-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-384"><a href="#cb17-384" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter the length of the sequences</span></span>
<span id="cb17-385"><a href="#cb17-385" aria-hidden="true" tabindex="-1"></a><span class="co"># (suggestion: select that length range accounting for many sequences)</span></span>
<span id="cb17-386"><a href="#cb17-386" aria-hidden="true" tabindex="-1"></a>seqtab.nochim <span class="ot">&lt;-</span> seqtab.nochim[, <span class="fu">nchar</span>(<span class="fu">colnames</span>(seqtab.nochim)) <span class="sc">%in%</span> <span class="fu">seq</span>(<span class="dv">402</span>, <span class="dv">428</span>)]</span>
<span id="cb17-387"><a href="#cb17-387" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-388"><a href="#cb17-388" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-389"><a href="#cb17-389" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 9: Check the number of sequences retained in each step of the processing</span></span>
<span id="cb17-392"><a href="#cb17-392" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-393"><a href="#cb17-393" aria-hidden="true" tabindex="-1"></a><span class="co"># IMPORTANT: this step can be made just in case we have one seqtab (one sequencing run)</span></span>
<span id="cb17-394"><a href="#cb17-394" aria-hidden="true" tabindex="-1"></a><span class="co"># Otherwise, skip this to the step 10)</span></span>
<span id="cb17-395"><a href="#cb17-395" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-396"><a href="#cb17-396" aria-hidden="true" tabindex="-1"></a>getN <span class="ot">&lt;-</span> <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">getUniques</span>(x))</span>
<span id="cb17-397"><a href="#cb17-397" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-398"><a href="#cb17-398" aria-hidden="true" tabindex="-1"></a>track <span class="ot">&lt;-</span> <span class="fu">cbind</span>(out, <span class="fu">sapply</span>(dadaFs, getN), <span class="fu">sapply</span>(dadaRs, getN), <span class="fu">sapply</span>(mergers, getN), <span class="fu">rowSums</span>(seqtab.nochim))</span>
<span id="cb17-399"><a href="#cb17-399" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(track) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"input"</span>, <span class="st">"filtered"</span>, <span class="st">"denoisedF"</span>, <span class="st">"denoisedR"</span>, <span class="st">"merged"</span>, <span class="st">"nonchim"</span>)</span>
<span id="cb17-400"><a href="#cb17-400" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(track) <span class="ot">&lt;-</span> sample.names</span>
<span id="cb17-401"><a href="#cb17-401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-402"><a href="#cb17-402" aria-hidden="true" tabindex="-1"></a><span class="co"># get the percentages</span></span>
<span id="cb17-403"><a href="#cb17-403" aria-hidden="true" tabindex="-1"></a>track <span class="ot">&lt;-</span> track <span class="sc">%&gt;%</span></span>
<span id="cb17-404"><a href="#cb17-404" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-405"><a href="#cb17-405" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb17-406"><a href="#cb17-406" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.filtered =</span> filtered <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> input,</span>
<span id="cb17-407"><a href="#cb17-407" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.denoisedF =</span> denoisedF <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> filtered,</span>
<span id="cb17-408"><a href="#cb17-408" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.denoisedR =</span> denoisedR <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> filtered,</span>
<span id="cb17-409"><a href="#cb17-409" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.merged =</span> merged <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> filtered,</span>
<span id="cb17-410"><a href="#cb17-410" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.nonchim =</span> nonchim <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> merged,</span>
<span id="cb17-411"><a href="#cb17-411" aria-hidden="true" tabindex="-1"></a>    <span class="at">Perc.retained =</span> nonchim <span class="sc">*</span> <span class="dv">100</span> <span class="sc">/</span> input</span>
<span id="cb17-412"><a href="#cb17-412" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb17-413"><a href="#cb17-413" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(track) <span class="co"># here we could see the % of the sequences obtained after each step</span></span>
<span id="cb17-414"><a href="#cb17-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-415"><a href="#cb17-415" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(track, <span class="at">file =</span> <span class="st">"Sequencing_quality.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">row.names =</span> T, <span class="at">col.names =</span> T)</span>
<span id="cb17-416"><a href="#cb17-416" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-417"><a href="#cb17-417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-418"><a href="#cb17-418" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 10: Taxonomical classification</span></span>
<span id="cb17-419"><a href="#cb17-419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-422"><a href="#cb17-422" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-423"><a href="#cb17-423" aria-hidden="true" tabindex="-1"></a>taxa_rdp <span class="ot">&lt;-</span> <span class="fu">assignTaxonomy</span>(seqtab.nochim, <span class="st">"/mnt/datos/databases/rdp_train_set_18_H.fa"</span>, <span class="at">multithread =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-424"><a href="#cb17-424" aria-hidden="true" tabindex="-1"></a><span class="co"># indicate the path where the database is located.</span></span>
<span id="cb17-425"><a href="#cb17-425" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-426"><a href="#cb17-426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-427"><a href="#cb17-427" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 11: Obtaining the ASV table</span></span>
<span id="cb17-430"><a href="#cb17-430" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-431"><a href="#cb17-431" aria-hidden="true" tabindex="-1"></a><span class="co"># Format the data</span></span>
<span id="cb17-432"><a href="#cb17-432" aria-hidden="true" tabindex="-1"></a>ASV <span class="ot">&lt;-</span> seqtab.nochim</span>
<span id="cb17-433"><a href="#cb17-433" aria-hidden="true" tabindex="-1"></a>ASVt <span class="ot">&lt;-</span> <span class="fu">t</span>(ASV)</span>
<span id="cb17-434"><a href="#cb17-434" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-435"><a href="#cb17-435" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace NA values by "unclassified" and remove the column "species" created during the classification</span></span>
<span id="cb17-436"><a href="#cb17-436" aria-hidden="true" tabindex="-1"></a>taxa_rdp_na <span class="ot">&lt;-</span> <span class="fu">apply</span>(taxa_rdp, <span class="dv">2</span>, tidyr<span class="sc">::</span>replace_na, <span class="st">"unclassified"</span>)[, <span class="sc">-</span><span class="dv">7</span>]</span>
<span id="cb17-437"><a href="#cb17-437" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-438"><a href="#cb17-438" aria-hidden="true" tabindex="-1"></a><span class="co"># Rename the ASVs (if we have 100 ASVs, we will name them from ASV001 to ASV100)</span></span>
<span id="cb17-439"><a href="#cb17-439" aria-hidden="true" tabindex="-1"></a>number.digit <span class="ot">&lt;-</span> <span class="fu">nchar</span>(<span class="fu">as.integer</span>(<span class="fu">nrow</span>(ASVt)))</span>
<span id="cb17-440"><a href="#cb17-440" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"ASV%0"</span>, number.digit, <span class="st">"d"</span>) <span class="co"># As many 0 as digits</span></span>
<span id="cb17-441"><a href="#cb17-441" aria-hidden="true" tabindex="-1"></a>ASV_names <span class="ot">&lt;-</span> <span class="fu">sprintf</span>(names, <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(ASVt))</span>
<span id="cb17-442"><a href="#cb17-442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-443"><a href="#cb17-443" aria-hidden="true" tabindex="-1"></a><span class="co"># Join taxonomy and ASV table</span></span>
<span id="cb17-444"><a href="#cb17-444" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">as.data.frame</span>(taxa_rdp_na, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>), <span class="fu">as.data.frame</span>(ASV_names, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>), <span class="fu">as.data.frame</span>(ASVt, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>))</span>
<span id="cb17-445"><a href="#cb17-445" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-446"><a href="#cb17-446" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the sequence of each ASV to the ASV table in the correct position</span></span>
<span id="cb17-447"><a href="#cb17-447" aria-hidden="true" tabindex="-1"></a>ASV_seqs <span class="ot">&lt;-</span> <span class="fu">rownames</span>(ASV_table_classified_raw)</span>
<span id="cb17-448"><a href="#cb17-448" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(ASV_table_classified_raw) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-449"><a href="#cb17-449" aria-hidden="true" tabindex="-1"></a>ASV_table_classified_raw <span class="ot">&lt;-</span> <span class="fu">cbind</span>(ASV_seqs, ASV_table_classified_raw)</span>
<span id="cb17-450"><a href="#cb17-450" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-451"><a href="#cb17-451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-452"><a href="#cb17-452" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 12: Mock community quality control</span></span>
<span id="cb17-455"><a href="#cb17-455" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-456"><a href="#cb17-456" aria-hidden="true" tabindex="-1"></a><span class="co"># Eliminacion de la MOCK---------------------------------</span></span>
<span id="cb17-457"><a href="#cb17-457" aria-hidden="true" tabindex="-1"></a>MockCommunity <span class="ot">&lt;-</span> <span class="cf">function</span>(data, MOCK_composition, ASV_column, <span class="at">choose.first =</span> <span class="cn">FALSE</span>) {</span>
<span id="cb17-458"><a href="#cb17-458" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get total number of sequences for each MOCK ASV</span></span>
<span id="cb17-459"><a href="#cb17-459" aria-hidden="true" tabindex="-1"></a>  sum_ASVs_MOCK <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(data[, <span class="fu">grep</span>(<span class="st">"MOCK"</span>, <span class="fu">colnames</span>(data)), <span class="at">drop =</span> <span class="cn">FALSE</span>])</span>
<span id="cb17-460"><a href="#cb17-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-461"><a href="#cb17-461" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Bind it to ASV_table</span></span>
<span id="cb17-462"><a href="#cb17-462" aria-hidden="true" tabindex="-1"></a>  ASV_table_counts_MOCK <span class="ot">&lt;-</span> <span class="fu">cbind</span>(data, sum_ASVs_MOCK)</span>
<span id="cb17-463"><a href="#cb17-463" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(ASV_table_counts_MOCK)[<span class="fu">ncol</span>(ASV_table_counts_MOCK)] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Total counts MOCK"</span>)</span>
<span id="cb17-464"><a href="#cb17-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-465"><a href="#cb17-465" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sort ASV table according to MOCK ASV rel.abundance</span></span>
<span id="cb17-466"><a href="#cb17-466" aria-hidden="true" tabindex="-1"></a>  ASV_table_counts_MOCK_sorted <span class="ot">&lt;-</span> ASV_table_counts_MOCK[<span class="fu">order</span>(ASV_table_counts_MOCK<span class="sc">$</span><span class="st">"Total counts MOCK"</span>, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), ]</span>
<span id="cb17-467"><a href="#cb17-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-468"><a href="#cb17-468" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Calculate percentage</span></span>
<span id="cb17-469"><a href="#cb17-469" aria-hidden="true" tabindex="-1"></a>  percentage <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-470"><a href="#cb17-470" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(ASV_table_counts_MOCK_sorted)) {</span>
<span id="cb17-471"><a href="#cb17-471" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">isTRUE</span>(<span class="fu">any</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span>Genus[i] <span class="sc">%in%</span> MOCK_composition[, <span class="dv">1</span>]))) { <span class="co"># for each line, if Genus is equal to any of the MOCK members,continue with the next line (next)</span></span>
<span id="cb17-472"><a href="#cb17-472" aria-hidden="true" tabindex="-1"></a>      <span class="cf">next</span></span>
<span id="cb17-473"><a href="#cb17-473" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb17-474"><a href="#cb17-474" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (<span class="fu">isTRUE</span>(choose.first)) {</span>
<span id="cb17-475"><a href="#cb17-475" aria-hidden="true" tabindex="-1"></a>        percentage <span class="ot">&lt;-</span> (ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span>)) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb17-476"><a href="#cb17-476" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(</span>
<span id="cb17-477"><a href="#cb17-477" aria-hidden="true" tabindex="-1"></a>          <span class="st">"First ASV found that does not belong to the MOCK community! It is"</span>, ASV_table_counts_MOCK_sorted[i, ][[ASV_column]], <span class="st">"which classifies as"</span>, ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span>Genus, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"and represents a"</span>,</span>
<span id="cb17-478"><a href="#cb17-478" aria-hidden="true" tabindex="-1"></a>          <span class="fu">round</span>((ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span>)) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">6</span>),</span>
<span id="cb17-479"><a href="#cb17-479" aria-hidden="true" tabindex="-1"></a>          <span class="st">"perc. of the sequences. This ASV was used to calculate the percentage"</span></span>
<span id="cb17-480"><a href="#cb17-480" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb17-481"><a href="#cb17-481" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span></span>
<span id="cb17-482"><a href="#cb17-482" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb17-483"><a href="#cb17-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-484"><a href="#cb17-484" aria-hidden="true" tabindex="-1"></a>      <span class="co"># if it finds a ASV which does not belong to the MOCK community, make a question to user.</span></span>
<span id="cb17-485"><a href="#cb17-485" aria-hidden="true" tabindex="-1"></a>      answer <span class="ot">&lt;-</span> <span class="fu">readline</span>(<span class="at">prompt =</span> <span class="fu">cat</span>(</span>
<span id="cb17-486"><a href="#cb17-486" aria-hidden="true" tabindex="-1"></a>        ASV_table_counts_MOCK_sorted[i, ][[ASV_column]],</span>
<span id="cb17-487"><a href="#cb17-487" aria-hidden="true" tabindex="-1"></a>        <span class="st">"does not belong to the MOCK community."</span>,</span>
<span id="cb17-488"><a href="#cb17-488" aria-hidden="true" tabindex="-1"></a>        <span class="st">"It representes a "</span>,</span>
<span id="cb17-489"><a href="#cb17-489" aria-hidden="true" tabindex="-1"></a>        <span class="fu">round</span>((ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span>)) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">6</span>),</span>
<span id="cb17-490"><a href="#cb17-490" aria-hidden="true" tabindex="-1"></a>        <span class="st">" perc. of the sequences"</span>, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"and it classifies as"</span>, ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span>Genus, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"Do you want to use this ASV to calculate the percentage?"</span>, <span class="st">"[answer yes or no]"</span></span>
<span id="cb17-491"><a href="#cb17-491" aria-hidden="true" tabindex="-1"></a>      ))</span>
<span id="cb17-492"><a href="#cb17-492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-493"><a href="#cb17-493" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (answer <span class="sc">==</span> <span class="st">"no"</span>) { <span class="co"># if the user chooses not to use the first spurious ASV, go to the next one</span></span>
<span id="cb17-494"><a href="#cb17-494" aria-hidden="true" tabindex="-1"></a>        <span class="cf">next</span></span>
<span id="cb17-495"><a href="#cb17-495" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb17-496"><a href="#cb17-496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-497"><a href="#cb17-497" aria-hidden="true" tabindex="-1"></a>      <span class="cf">if</span> (answer <span class="sc">==</span> <span class="st">"yes"</span>) { <span class="co"># when the user says "yes", store de percentage and print the ASV name, classification and %</span></span>
<span id="cb17-498"><a href="#cb17-498" aria-hidden="true" tabindex="-1"></a>        percentage <span class="ot">&lt;-</span> (ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span>)) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb17-499"><a href="#cb17-499" aria-hidden="true" tabindex="-1"></a>        <span class="fu">cat</span>(</span>
<span id="cb17-500"><a href="#cb17-500" aria-hidden="true" tabindex="-1"></a>          <span class="st">"You made a decision!"</span>, ASV_table_counts_MOCK_sorted[i, ][[ASV_column]], <span class="st">"which classifies as"</span>, ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span>Genus, <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>, <span class="st">"and represents a"</span>,</span>
<span id="cb17-501"><a href="#cb17-501" aria-hidden="true" tabindex="-1"></a>          <span class="fu">round</span>((ASV_table_counts_MOCK_sorted[i, ]<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span> <span class="sc">/</span> <span class="fu">sum</span>(ASV_table_counts_MOCK_sorted<span class="sc">$</span><span class="st">`</span><span class="at">Total counts MOCK</span><span class="st">`</span>)) <span class="sc">*</span> <span class="dv">100</span>, <span class="at">digits =</span> <span class="dv">6</span>),</span>
<span id="cb17-502"><a href="#cb17-502" aria-hidden="true" tabindex="-1"></a>          <span class="st">"perc. of the sequences, was used to calculate the percentage"</span></span>
<span id="cb17-503"><a href="#cb17-503" aria-hidden="true" tabindex="-1"></a>        )</span>
<span id="cb17-504"><a href="#cb17-504" aria-hidden="true" tabindex="-1"></a>        <span class="cf">break</span></span>
<span id="cb17-505"><a href="#cb17-505" aria-hidden="true" tabindex="-1"></a>      } <span class="cf">else</span> {</span>
<span id="cb17-506"><a href="#cb17-506" aria-hidden="true" tabindex="-1"></a>        <span class="fu">stop</span>(<span class="st">"Error: Answers have to fit  'yes' or 'no'"</span>)</span>
<span id="cb17-507"><a href="#cb17-507" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb17-508"><a href="#cb17-508" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb17-509"><a href="#cb17-509" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb17-510"><a href="#cb17-510" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-511"><a href="#cb17-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-512"><a href="#cb17-512" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Remove MOCK columns</span></span>
<span id="cb17-513"><a href="#cb17-513" aria-hidden="true" tabindex="-1"></a>  ASV_table_without_MOCK <span class="ot">&lt;-</span> ASV_table_counts_MOCK_sorted[, <span class="sc">-</span><span class="fu">grep</span>(<span class="st">"MOCK"</span>, <span class="fu">colnames</span>(ASV_table_counts_MOCK_sorted))]</span>
<span id="cb17-514"><a href="#cb17-514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-515"><a href="#cb17-515" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get number of sequences of each ASV without MOCK</span></span>
<span id="cb17-516"><a href="#cb17-516" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(ASV_table_without_MOCK) <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb17-517"><a href="#cb17-517" aria-hidden="true" tabindex="-1"></a>  ASV_sums <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(ASV_table_without_MOCK[, <span class="dv">9</span><span class="sc">:</span><span class="fu">ncol</span>(ASV_table_without_MOCK)])</span>
<span id="cb17-518"><a href="#cb17-518" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Get total number of sequences</span></span>
<span id="cb17-519"><a href="#cb17-519" aria-hidden="true" tabindex="-1"></a>  sum.total <span class="ot">&lt;-</span> <span class="fu">sum</span>(ASV_sums)</span>
<span id="cb17-520"><a href="#cb17-520" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply percentage to sequence number</span></span>
<span id="cb17-521"><a href="#cb17-521" aria-hidden="true" tabindex="-1"></a>  nseq_cutoff <span class="ot">&lt;-</span> (percentage <span class="sc">/</span> <span class="dv">100</span>) <span class="sc">*</span> sum.total</span>
<span id="cb17-522"><a href="#cb17-522" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter table.</span></span>
<span id="cb17-523"><a href="#cb17-523" aria-hidden="true" tabindex="-1"></a>  ASV_filtered <span class="ot">&lt;-</span> ASV_table_without_MOCK[<span class="fu">which</span>(ASV_sums <span class="sc">&gt;</span> nseq_cutoff), ]</span>
<span id="cb17-524"><a href="#cb17-524" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-525"><a href="#cb17-525" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-526"><a href="#cb17-526" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Sort table in ascending order of ASV names</span></span>
<span id="cb17-527"><a href="#cb17-527" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_sorted <span class="ot">&lt;-</span> ASV_filtered[<span class="fu">order</span>(ASV_filtered[[ASV_column]]), ]</span>
<span id="cb17-528"><a href="#cb17-528" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(ASV_filtered_sorted)</span>
<span id="cb17-529"><a href="#cb17-529" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-530"><a href="#cb17-530" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-531"><a href="#cb17-531" aria-hidden="true" tabindex="-1"></a>ASV_filtered_MOCK <span class="ot">&lt;-</span> <span class="fu">MockCommunity</span>(ASV_table_classified_raw, mock_composition, <span class="at">ASV_column =</span> <span class="st">"ASV_names"</span>)</span>
<span id="cb17-532"><a href="#cb17-532" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-533"><a href="#cb17-533" aria-hidden="true" tabindex="-1"></a><span class="co"># select the relative abundance of the first ASV that is not included in the Mock Community. That percentage will be considered the detection limit of sequencing</span></span>
<span id="cb17-534"><a href="#cb17-534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-535"><a href="#cb17-535" aria-hidden="true" tabindex="-1"></a><span class="co"># IMPORTANT: write down and keep this cut-off value if you will work also with ITS2 or fungal reads, because you will work just with same cut-off</span></span>
<span id="cb17-536"><a href="#cb17-536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-537"><a href="#cb17-537" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-538"><a href="#cb17-538" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-539"><a href="#cb17-539" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 13: Removal of Plastids, Plants, Chloroplast and other artifcats sequences</span></span>
<span id="cb17-542"><a href="#cb17-542" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-543"><a href="#cb17-543" aria-hidden="true" tabindex="-1"></a>ASV_final <span class="ot">&lt;-</span> ASV_filtered_MOCK[(<span class="fu">which</span>(ASV_filtered_MOCK<span class="sc">$</span>Genus <span class="sc">!=</span> <span class="st">"Streptophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb17-544"><a href="#cb17-544" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Genus <span class="sc">!=</span> <span class="st">"Chlorophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb17-545"><a href="#cb17-545" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Genus <span class="sc">!=</span> <span class="st">"Bacillariophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb17-546"><a href="#cb17-546" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Family <span class="sc">!=</span> <span class="st">"Streptophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb17-547"><a href="#cb17-547" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Family <span class="sc">!=</span> <span class="st">"Chlorophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb17-548"><a href="#cb17-548" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Family <span class="sc">!=</span> <span class="st">"Bacillariophyta"</span> <span class="sc">&amp;</span></span>
<span id="cb17-549"><a href="#cb17-549" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Family <span class="sc">!=</span> <span class="st">"Mitochondria"</span> <span class="sc">&amp;</span></span>
<span id="cb17-550"><a href="#cb17-550" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Class <span class="sc">!=</span> <span class="st">"Chloroplast"</span> <span class="sc">&amp;</span></span>
<span id="cb17-551"><a href="#cb17-551" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Order <span class="sc">!=</span> <span class="st">"Chloroplast"</span> <span class="sc">&amp;</span></span>
<span id="cb17-552"><a href="#cb17-552" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Kingdom <span class="sc">!=</span> <span class="st">"Eukaryota"</span> <span class="sc">&amp;</span></span>
<span id="cb17-553"><a href="#cb17-553" aria-hidden="true" tabindex="-1"></a>  ASV_filtered_MOCK<span class="sc">$</span>Kingdom <span class="sc">!=</span> <span class="st">"unclassified"</span>)), ]</span>
<span id="cb17-554"><a href="#cb17-554" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-555"><a href="#cb17-555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-556"><a href="#cb17-556" aria-hidden="true" tabindex="-1"></a><span class="fu"># Step 14: Refinement of the taxonomy</span></span>
<span id="cb17-557"><a href="#cb17-557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-560"><a href="#cb17-560" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb17-561"><a href="#cb17-561" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the taxonomy of that ASVs classified at Phylum level as "Cyanobacteria/Chloroplast"</span></span>
<span id="cb17-562"><a href="#cb17-562" aria-hidden="true" tabindex="-1"></a><span class="fu">View</span>(ASV_final[<span class="fu">which</span>(ASV_final<span class="sc">$</span>Phylum <span class="sc">==</span> <span class="st">"Cyanobacteria/Chloroplast"</span>), ])</span>
<span id="cb17-563"><a href="#cb17-563" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-564"><a href="#cb17-564" aria-hidden="true" tabindex="-1"></a><span class="co"># Suggestion: if they are classified at Class level properly (as a Cyanobacteria), keep them</span></span>
<span id="cb17-565"><a href="#cb17-565" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-566"><a href="#cb17-566" aria-hidden="true" tabindex="-1"></a>ASV_final2 <span class="ot">&lt;-</span> ASV_final[(<span class="fu">which</span>(ASV_final<span class="sc">$</span>Phylum <span class="sc">!=</span> <span class="st">"Cyanobacteria/Chloroplast"</span> <span class="sc">|</span> ASV_final<span class="sc">$</span>Class <span class="sc">!=</span> <span class="st">"unclassified"</span>)), ]</span>
<span id="cb17-567"><a href="#cb17-567" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-568"><a href="#cb17-568" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the sequences of those ASV classified as "Cyanobacteria/Chloroplast"</span></span>
<span id="cb17-569"><a href="#cb17-569" aria-hidden="true" tabindex="-1"></a>cianos <span class="ot">&lt;-</span> ASV_final2[<span class="fu">which</span>(ASV_final2<span class="sc">$</span>Phylum <span class="sc">==</span> <span class="st">"Cyanobacteria/Chloroplast"</span>), ]</span>
<span id="cb17-570"><a href="#cb17-570" aria-hidden="true" tabindex="-1"></a>seq_cianos <span class="ot">&lt;-</span> <span class="fu">as.list</span>(cianos<span class="sc">$</span>ASV_seqs)</span>
<span id="cb17-571"><a href="#cb17-571" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_cianos, <span class="at">names =</span> cianos<span class="sc">$</span>ASV_names, <span class="at">file.out =</span> <span class="st">"cianos.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span> <span class="dv">1000</span>, <span class="at">as.string =</span> <span class="cn">FALSE</span>)</span>
<span id="cb17-572"><a href="#cb17-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-573"><a href="#cb17-573" aria-hidden="true" tabindex="-1"></a><span class="co"># align the corresponding sequences by BLAST and compare them with those held in the nt database of NCBI</span></span>
<span id="cb17-574"><a href="#cb17-574" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>((<span class="st">"/home/programs/ncbi-blast-2.11.0+/bin/blastn -query cianos.fas -db /mnt/datos/databases/nt -out cianos_hits.txt -outfmt '6 std stitle' -show_gis -max_target_seqs 20 -parse_deflines -num_threads 10"</span>), <span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-575"><a href="#cb17-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-576"><a href="#cb17-576" aria-hidden="true" tabindex="-1"></a><span class="co"># (we verified that all the sequences corresponded to real cyanobacteria, so we did not remove them from the dataset)</span></span>
<span id="cb17-577"><a href="#cb17-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-578"><a href="#cb17-578" aria-hidden="true" tabindex="-1"></a><span class="co"># Repeat the same with sequences not classified at Phylum level, since they can be artifacts,</span></span>
<span id="cb17-579"><a href="#cb17-579" aria-hidden="true" tabindex="-1"></a><span class="co"># even sequences of the plant host (specially if you are working with endosphere or plant tissues)</span></span>
<span id="cb17-580"><a href="#cb17-580" aria-hidden="true" tabindex="-1"></a>unclassified_phyla <span class="ot">&lt;-</span> ASV_final2[<span class="fu">which</span>(ASV_final2<span class="sc">$</span>Phylum <span class="sc">==</span> <span class="st">"unclassified"</span>), ]</span>
<span id="cb17-581"><a href="#cb17-581" aria-hidden="true" tabindex="-1"></a>seq_unclassified_phyla <span class="ot">&lt;-</span> <span class="fu">as.list</span>(unclassified_phyla<span class="sc">$</span>ASV_seqs)</span>
<span id="cb17-582"><a href="#cb17-582" aria-hidden="true" tabindex="-1"></a><span class="fu">write.fasta</span>(seq_unclassified_phyla, <span class="at">names =</span> unclassified_phyla<span class="sc">$</span>ASV_names, <span class="at">file.out =</span> <span class="st">"unclassified_phyla.fas"</span>, <span class="at">open =</span> <span class="st">"w"</span>, <span class="at">nbchar =</span> <span class="dv">1000</span>, <span class="at">as.string =</span> <span class="cn">FALSE</span>)</span>
<span id="cb17-583"><a href="#cb17-583" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-584"><a href="#cb17-584" aria-hidden="true" tabindex="-1"></a><span class="co"># BLAST</span></span>
<span id="cb17-585"><a href="#cb17-585" aria-hidden="true" tabindex="-1"></a><span class="fu">system</span>((<span class="st">"/home/programs/ncbi-blast-2.11.0+/bin/blastn -query unclassified_phyla.fas -db /mnt/datos/databases/nt -out unclassified_phyla_hits.txt -outfmt '6 std stitle' -show_gis -max_target_seqs 20 -parse_deflines -num_threads 10"</span>), <span class="at">intern =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-586"><a href="#cb17-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-587"><a href="#cb17-587" aria-hidden="true" tabindex="-1"></a><span class="co"># (we detected that the ASV03686 corresponded to plant host, so we removed it)</span></span>
<span id="cb17-588"><a href="#cb17-588" aria-hidden="true" tabindex="-1"></a>ASV_final_all_filters <span class="ot">&lt;-</span> ASV_final2[(<span class="fu">which</span>(ASV_final2<span class="sc">$</span>Phylum <span class="sc">!=</span> <span class="st">"Cyanobacteria/Chloroplast"</span>)), ]</span>
<span id="cb17-589"><a href="#cb17-589" aria-hidden="true" tabindex="-1"></a>ASV_final_all_filters <span class="ot">&lt;-</span> ASV_final_all_filters[(<span class="fu">which</span>(ASV_final_all_filters<span class="sc">$</span>ASV_names <span class="sc">!=</span> <span class="st">"ASV03686"</span>)), ]</span>
<span id="cb17-590"><a href="#cb17-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-591"><a href="#cb17-591" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the definitive ASV table</span></span>
<span id="cb17-592"><a href="#cb17-592" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(ASV_final_all_filters, <span class="at">file =</span> <span class="st">"ASV_final.txt"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb17-593"><a href="#cb17-593" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb17-594"><a href="#cb17-594" aria-hidden="true" tabindex="-1"></a></span>
</code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>